// ng-uikit-pro-standard/esm5/lib/free/tooltip/tooltip.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TooltipContainerComponent } from './tooltip.component';
import { TooltipDirective } from './tooltip.directive';
import { TooltipConfig } from './tooltip.service';
import { ComponentLoaderFactory } from '../utils/component-loader/component-loader.factory';
import { PositioningService } from '../utils/positioning/positioning.service';
var TooltipModule = /** @class */ (function () {
    function TooltipModule() {
    }
    TooltipModule_1 = TooltipModule;
    TooltipModule.forRoot = function () {
        return {
            ngModule: TooltipModule_1,
            providers: [TooltipConfig, ComponentLoaderFactory, PositioningService],
        };
    };
    var TooltipModule_1;
    TooltipModule = TooltipModule_1 = __decorate([
        NgModule({
            imports: [CommonModule],
            declarations: [TooltipDirective, TooltipContainerComponent],
            exports: [TooltipDirective],
            entryComponents: [TooltipContainerComponent],
        })
    ], TooltipModule);
    return TooltipModule;
}());
export { TooltipModule };
