// ng-uikit-pro-standard/esm5/lib/pro/auto-format/auto-format.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { MdbDateFormatDirective } from './mdb-date-format.directive';
import { MdbCreditCardDirective } from './mdb-credit-card.directive';
import { MdbCvvDirective } from './mdb-cvv.directive';
import * as ɵngcc0 from '@angular/core';
var AutoFormatModule = /** @class */ (function () {
    function AutoFormatModule() {
    }
AutoFormatModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AutoFormatModule });
AutoFormatModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AutoFormatModule_Factory(t) { return new (t || AutoFormatModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AutoFormatModule, { declarations: function () { return [MdbDateFormatDirective,
        MdbCreditCardDirective,
        MdbCvvDirective]; }, exports: function () { return [MdbDateFormatDirective,
        MdbCreditCardDirective,
        MdbCvvDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AutoFormatModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    MdbDateFormatDirective,
                    MdbCreditCardDirective,
                    MdbCvvDirective
                ],
                exports: [
                    MdbDateFormatDirective,
                    MdbCreditCardDirective,
                    MdbCvvDirective
                ]
            }]
    }], function () { return []; }, null); })();
    return AutoFormatModule;
}());
export { AutoFormatModule };
