// ng-uikit-pro-standard/esm2015/lib/free/cards/mdb-card-text.component.js
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class MdbCardTextComponent {
}
MdbCardTextComponent.ɵfac = function MdbCardTextComponent_Factory(t) { return new (t || MdbCardTextComponent)(); };
MdbCardTextComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MdbCardTextComponent, selectors: [["mdb-card-text"]], inputs: { class: "class" }, ngContentSelectors: _c0, decls: 2, vars: 3, template: function MdbCardTextComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "p");
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("card-text ", ctx.class, " ");
    } }, encapsulation: 2, changeDetection: 0 });
MdbCardTextComponent.propDecorators = {
    class: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbCardTextComponent, [{
        type: Component,
        args: [{
                selector: 'mdb-card-text',
                template: "<p class=\"card-text {{class}} \">\n    <ng-content></ng-content>\n</p>",
                changeDetection: ChangeDetectionStrategy.OnPush
            }]
    }], null, { class: [{
            type: Input
        }] }); })();
