// ng-uikit-pro-standard/esm5/lib/pro/scroll-spy/scroll-spy.directive.js
import { __decorate, __metadata } from "tslib";
import { Component, ContentChildren, EventEmitter, Input, Output, QueryList, ViewEncapsulation, } from '@angular/core';
import { ScrollSpyLinkDirective } from './scroll-spy-link.directive';
import { ScrollSpyService } from './scroll-spy.service';
import { distinctUntilChanged } from 'rxjs/operators';
var ScrollSpyDirective = /** @class */ (function () {
    function ScrollSpyDirective(scrollSpyService) {
        this.scrollSpyService = scrollSpyService;
        this.activeLinkChange = new EventEmitter();
    }
    Object.defineProperty(ScrollSpyDirective.prototype, "id", {
        get: function () {
            return this._id;
        },
        set: function (newId) {
            if (newId) {
                this._id = newId;
            }
        },
        enumerable: true,
        configurable: true
    });
    ScrollSpyDirective.prototype.ngOnInit = function () {
        var _this = this;
        this.activeSub = this.scrollSpyService.active$
            .pipe(distinctUntilChanged())
            .subscribe(function (activeLink) {
            _this.activeLinkChange.emit(activeLink);
        });
    };
    ScrollSpyDirective.prototype.ngAfterContentInit = function () {
        this.scrollSpyService.addScrollSpy({ id: this.id, links: this.links });
    };
    ScrollSpyDirective.prototype.ngOnDestroy = function () {
        this.scrollSpyService.removeScrollSpy(this.id);
        this.activeSub.unsubscribe();
    };
    ScrollSpyDirective.ctorParameters = function () { return [
        { type: ScrollSpyService }
    ]; };
    __decorate([
        ContentChildren(ScrollSpyLinkDirective, { descendants: true }),
        __metadata("design:type", QueryList)
    ], ScrollSpyDirective.prototype, "links", void 0);
    __decorate([
        Input('mdbScrollSpy'),
        __metadata("design:type", String),
        __metadata("design:paramtypes", [String])
    ], ScrollSpyDirective.prototype, "id", null);
    __decorate([
        Output(),
        __metadata("design:type", EventEmitter)
    ], ScrollSpyDirective.prototype, "activeLinkChange", void 0);
    ScrollSpyDirective = __decorate([
        Component({
            // tslint:disable-next-line:component-selector
            selector: '[mdbScrollSpy]',
            template: '<ng-content></ng-content>',
            encapsulation: ViewEncapsulation.None,
            styles: [".nav-pills.default-pills .nav-item a{text-align:left;color:#666;border-radius:0}.nav-pills.default-pills .nav-item a:hover{background-color:transparent;color:#45526e;border-left:.0625rem solid #45526e;font-weight:500}.nav-pills.default-pills .nav-item a.active{background-color:transparent;color:#45526e;border-left:.125rem solid #45526e;box-shadow:none;font-weight:500}.nav-pills.default-pills .nav-item a.active:active,.nav-pills.default-pills .nav-item a.active:focus,.nav-pills.default-pills .nav-item a.active:hover{background-color:transparent;color:#45526e;font-weight:500}.nav-pills .nav-item+.nav-item{margin-left:0}@media only screen and (max-width:991px){.sticky,.sticky-placeholder{display:none}}#scrollspy{width:100%}@media only screen and (max-width:992px){.col-lg-4 .sticky,.col-md-4 .sticky,.col-xl-4 .sticky{width:12.5rem}}@media only screen and (min-width:992px){.col-md-4 .sticky .col-lg-4 .sticky,.col-xl-4 .sticky{width:25rem}}@media only screen and (min-width:1200px){.col-md-4 .sticky{width:18.75rem}}@media only screen and (min-width:1440px){.col-md-4 .sticky{width:21.875rem}}.dotted-scrollspy{display:block;position:fixed;top:50%;right:0;transform:translateY(-50%);background:rgba(0,0,0,.55);border-radius:.125rem 0 0 .125rem}.dotted-scrollspy li{display:block;padding:0 1rem}.dotted-scrollspy li:first-child{padding-top:.625rem}.dotted-scrollspy li:last-child{padding-bottom:.625rem}.dotted-scrollspy li a{padding:.5rem}.dotted-scrollspy li a span{display:block;background-color:rgba(255,255,255,.54);width:.5rem;height:.5rem;border-radius:50%}.dotted-scrollspy li a.active span{background:#fff}.scrollspy-example{overflow-y:scroll;position:relative;height:12.5rem;padding:1rem}.flex-column .nav-link{padding:.5rem 1rem!important}"]
        })
        // tslint:disable-next-line:component-class-suffix
        ,
        __metadata("design:paramtypes", [ScrollSpyService])
    ], ScrollSpyDirective);
    return ScrollSpyDirective;
}());
export { ScrollSpyDirective };
