// ng-uikit-pro-standard/esm2015/lib/pro/option/option-group.component.js
import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input, HostBinding, Optional, Inject, } from '@angular/core';
import { MDB_OPTION_GROUP, MDB_OPTION_PARENT } from './option.component';
import * as ɵngcc0 from '@angular/core';

const _c0 = [[["mdb-select-option"]]];
const _c1 = ["mdb-select-option"];
export class OptionGroupComponent {
    constructor(_parent) {
        this._parent = _parent;
        this.optionGroup = true;
        this._optionHeight = 48;
        this._disabled = false;
    }
    get disabled() {
        return this._disabled;
    }
    set disabled(value) {
        this._disabled = value;
    }
    ngOnInit() {
        if (this._parent && this._parent.visibleOptions && this._parent.optionHeight) {
            this._optionHeight = this._parent.optionHeight;
        }
    }
    ngAfterContentInit() { }
}
OptionGroupComponent.ɵfac = function OptionGroupComponent_Factory(t) { return new (t || OptionGroupComponent)(ɵngcc0.ɵɵdirectiveInject(MDB_OPTION_PARENT, 8)); };
OptionGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: OptionGroupComponent, selectors: [["mdb-option-group"]], hostVars: 2, hostBindings: function OptionGroupComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("mdb-option-group", ctx.optionGroup);
    } }, inputs: { disabled: "disabled", label: "label" }, features: [ɵngcc0.ɵɵProvidersFeature([{ provide: MDB_OPTION_GROUP, useExisting: OptionGroupComponent }])], ngContentSelectors: _c1, decls: 3, vars: 5, consts: [[1, "mdb-option-group-label"]], template: function OptionGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "label", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(2);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("height", ctx._optionHeight, "px")("line-height", ctx._optionHeight, "px");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.label);
    } }, styles: [".mdb-option-group{display:flex;flex-direction:column}.mdb-option-group-label{border-top:1px solid #eee;color:#9e9e9e;margin:0;padding-left:16px;padding-right:16px;text-overflow:ellipsis;white-space:nowrap}.mdb-option-group:first-child .mdb-option-group-label{border-top:0}"], encapsulation: 2, changeDetection: 0 });
OptionGroupComponent.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [MDB_OPTION_PARENT,] }] }
];
OptionGroupComponent.propDecorators = {
    optionGroup: [{ type: HostBinding, args: ['class.mdb-option-group',] }],
    label: [{ type: Input }],
    disabled: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OptionGroupComponent, [{
        type: Component,
        args: [{
                selector: 'mdb-option-group',
                template: "<label\n  class=\"mdb-option-group-label\"\n  [style.height.px]=\"_optionHeight\"\n  [style.line-height.px]=\"_optionHeight\"\n  >{{ label }}</label\n>\n<ng-content select=\"mdb-select-option\"></ng-content>\n",
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                providers: [{ provide: MDB_OPTION_GROUP, useExisting: OptionGroupComponent }],
                styles: [".mdb-option-group{display:flex;flex-direction:column}.mdb-option-group-label{border-top:1px solid #eee;color:#9e9e9e;margin:0;padding-left:16px;padding-right:16px;text-overflow:ellipsis;white-space:nowrap}.mdb-option-group:first-child .mdb-option-group-label{border-top:0}"]
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [MDB_OPTION_PARENT]
            }] }]; }, { optionGroup: [{
            type: HostBinding,
            args: ['class.mdb-option-group']
        }], disabled: [{
            type: Input
        }], label: [{
            type: Input
        }] }); })();
