// ng-uikit-pro-standard/esm5/lib/free/utils/positioning/positioning.service.js
import { __decorate, __metadata, __param } from "tslib";
import { Injectable, ElementRef, RendererFactory2, Inject, PLATFORM_ID, NgZone, } from '@angular/core';
import { isPlatformBrowser } from '@angular/common';
import { positionElements } from './ng-positioning';
import { fromEvent, merge, of, animationFrameScheduler, Subject } from 'rxjs';
var PositioningService = /** @class */ (function () {
    function PositioningService(rendererFactory, platformId, _ngZone) {
        var _this = this;
        this._ngZone = _ngZone;
        this.update$$ = new Subject();
        this.positionElements = new Map();
        if (isPlatformBrowser(platformId)) {
            this._ngZone.runOutsideAngular(function () {
                merge(fromEvent(window, 'scroll'), fromEvent(window, 'resize'), 
                // tslint:disable-next-line: deprecation
                of(0, animationFrameScheduler), _this.update$$).subscribe(function () {
                    _this.positionElements.forEach(function (positionElement) {
                        positionElements(_getHtmlElement(positionElement.target), _getHtmlElement(positionElement.element), positionElement.attachment, positionElement.appendToBody, _this.options, rendererFactory.createRenderer(null, null));
                    });
                });
            });
        }
    }
    PositioningService.prototype.position = function (options) {
        this.addPositionElement(options);
    };
    PositioningService.prototype.addPositionElement = function (options) {
        this.positionElements.set(_getHtmlElement(options.element), options);
    };
    PositioningService.prototype.calcPosition = function () {
        this.update$$.next();
    };
    PositioningService.prototype.deletePositionElement = function (elRef) {
        this.positionElements.delete(_getHtmlElement(elRef));
    };
    PositioningService.prototype.setOptions = function (options) {
        this.options = options;
    };
    PositioningService.ctorParameters = function () { return [
        { type: RendererFactory2 },
        { type: Number, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] },
        { type: NgZone }
    ]; };
    PositioningService = __decorate([
        Injectable(),
        __param(1, Inject(PLATFORM_ID)),
        __metadata("design:paramtypes", [RendererFactory2, Number, NgZone])
    ], PositioningService);
    return PositioningService;
}());
export { PositioningService };
function _getHtmlElement(element) {
    // it means that we got a selector
    if (element && typeof element === 'string') {
        return document.querySelector(element);
    }
    if (element instanceof ElementRef) {
        return element.nativeElement;
    }
    return element;
}
