// ng-uikit-pro-standard/esm2015/lib/pro/file-input/module/mdb-uploader.module.js
import { NgModule } from '@angular/core';
import { MDBFileDropDirective } from '../directives/mdb-file-drop.directive';
import { MDBFileSelectDirective } from '../directives/mdb-file-select.directive';
import * as ɵngcc0 from '@angular/core';
export class FileInputModule {
}
FileInputModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FileInputModule });
FileInputModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FileInputModule_Factory(t) { return new (t || FileInputModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FileInputModule, { declarations: function () { return [MDBFileSelectDirective,
        MDBFileDropDirective]; }, exports: function () { return [MDBFileSelectDirective,
        MDBFileDropDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FileInputModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    MDBFileSelectDirective,
                    MDBFileDropDirective
                ],
                exports: [
                    MDBFileSelectDirective,
                    MDBFileDropDirective
                ]
            }]
    }], null, null); })();
