// ng-uikit-pro-standard/esm5/lib/pro/option/option.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { OptionComponent } from './option.component';
import { OptionGroupComponent } from './option-group.component';
import { CheckboxModule } from './../../free/checkbox/checkbox.module';
import { SelectAllOptionComponent } from './select-all-option';
import * as ɵngcc0 from '@angular/core';
var MdbOptionModule = /** @class */ (function () {
    function MdbOptionModule() {
    }
MdbOptionModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MdbOptionModule });
MdbOptionModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MdbOptionModule_Factory(t) { return new (t || MdbOptionModule)(); }, imports: [[CommonModule, CheckboxModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MdbOptionModule, { declarations: function () { return [OptionComponent, SelectAllOptionComponent, OptionGroupComponent]; }, imports: function () { return [CommonModule, CheckboxModule]; }, exports: function () { return [OptionComponent, OptionGroupComponent, SelectAllOptionComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbOptionModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, CheckboxModule],
                declarations: [OptionComponent, SelectAllOptionComponent, OptionGroupComponent],
                exports: [OptionComponent, OptionGroupComponent, SelectAllOptionComponent]
            }]
    }], function () { return []; }, null); })();
    return MdbOptionModule;
}());
export { MdbOptionModule };
