// ng-uikit-pro-standard/esm5/lib/pro/option/option-group.component.js
import { __decorate, __metadata, __param } from "tslib";
import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input, HostBinding, Optional, Inject, } from '@angular/core';
import { MDB_OPTION_GROUP, MDB_OPTION_PARENT } from './option.component';
var OptionGroupComponent = /** @class */ (function () {
    function OptionGroupComponent(_parent) {
        this._parent = _parent;
        this.optionGroup = true;
        this._optionHeight = 48;
        this._disabled = false;
    }
    OptionGroupComponent_1 = OptionGroupComponent;
    Object.defineProperty(OptionGroupComponent.prototype, "disabled", {
        get: function () {
            return this._disabled;
        },
        set: function (value) {
            this._disabled = value;
        },
        enumerable: true,
        configurable: true
    });
    OptionGroupComponent.prototype.ngOnInit = function () {
        if (this._parent && this._parent.visibleOptions && this._parent.optionHeight) {
            this._optionHeight = this._parent.optionHeight;
        }
    };
    OptionGroupComponent.prototype.ngAfterContentInit = function () { };
    var OptionGroupComponent_1;
    OptionGroupComponent.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [MDB_OPTION_PARENT,] }] }
    ]; };
    __decorate([
        HostBinding('class.mdb-option-group'),
        __metadata("design:type", Object)
    ], OptionGroupComponent.prototype, "optionGroup", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], OptionGroupComponent.prototype, "label", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean),
        __metadata("design:paramtypes", [Boolean])
    ], OptionGroupComponent.prototype, "disabled", null);
    OptionGroupComponent = OptionGroupComponent_1 = __decorate([
        Component({
            selector: 'mdb-option-group',
            template: "<label\n  class=\"mdb-option-group-label\"\n  [style.height.px]=\"_optionHeight\"\n  [style.line-height.px]=\"_optionHeight\"\n  >{{ label }}</label\n>\n<ng-content select=\"mdb-select-option\"></ng-content>\n",
            encapsulation: ViewEncapsulation.None,
            changeDetection: ChangeDetectionStrategy.OnPush,
            providers: [{ provide: MDB_OPTION_GROUP, useExisting: OptionGroupComponent_1 }],
            styles: [".mdb-option-group{display:flex;flex-direction:column}.mdb-option-group-label{white-space:nowrap;text-overflow:ellipsis;color:#9e9e9e;padding-left:16px;padding-right:16px;margin:0;border-top:1px solid #eee}.mdb-option-group:first-child .mdb-option-group-label{border-top:0}"]
        }),
        __param(0, Optional()), __param(0, Inject(MDB_OPTION_PARENT)),
        __metadata("design:paramtypes", [Object])
    ], OptionGroupComponent);
    return OptionGroupComponent;
}());
export { OptionGroupComponent };
