// ng-uikit-pro-standard/esm2015/lib/pro/progressbars/progress-bars-module/index.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressBarComponent } from './progressbar.component';
import * as ɵngcc0 from '@angular/core';
export { ProgressBarComponent } from './progressbar.component';
export class MdProgressBarModule {
    static forRoot() {
        return {
            ngModule: MdProgressBarModule,
            providers: [],
        };
    }
}
MdProgressBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MdProgressBarModule });
MdProgressBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MdProgressBarModule_Factory(t) { return new (t || MdProgressBarModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MdProgressBarModule, { declarations: function () { return [ProgressBarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ProgressBarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdProgressBarModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [ProgressBarComponent],
                declarations: [ProgressBarComponent]
            }]
    }], null, null); })();
