// ng-uikit-pro-standard/esm2015/lib/pro/inputs/char-counter.module.js
import { NgModule } from '@angular/core';
import { CharCounterDirective } from './char-counter.directive';
import * as ɵngcc0 from '@angular/core';
export class CharCounterModule {
    static forRoot() {
        return { ngModule: CharCounterModule, providers: [] };
    }
}
CharCounterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CharCounterModule });
CharCounterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CharCounterModule_Factory(t) { return new (t || CharCounterModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CharCounterModule, { declarations: function () { return [CharCounterDirective]; }, exports: function () { return [CharCounterDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CharCounterModule, [{
        type: NgModule,
        args: [{
                declarations: [CharCounterDirective],
                exports: [CharCounterDirective]
            }]
    }], null, null); })();
