// ng-uikit-pro-standard/esm5/lib/pro/tree-view/tree-view.component.js
import { __decorate, __metadata, __read, __values } from "tslib";
import { Component, Input, Output, EventEmitter, OnInit, ViewEncapsulation, ChangeDetectionStrategy, ChangeDetectorRef, HostBinding, } from '@angular/core';
import { SPACE, ENTER } from '../../free/utils/keyboard-navigation';
var MdbTreeComponent = /** @class */ (function () {
    function MdbTreeComponent(_cdRef) {
        this._cdRef = _cdRef;
        this.checked = new EventEmitter();
        this.checkedKeys = new EventEmitter();
        this.nodesChanged = new EventEmitter();
        this.checkboxes = false;
        this.toggleOnTitleClick = false;
        this._expandAll = false;
        this.checkedValues = [];
        this.toggle = {};
    }
    Object.defineProperty(MdbTreeComponent.prototype, "expandAll", {
        set: function (value) {
            if (this.nodes && this.nodes.entries()) {
                this._expandAll = value;
                this.toggleExpandAll();
            }
        },
        enumerable: true,
        configurable: true
    });
    MdbTreeComponent.prototype.ngOnInit = function () {
        if (this.nodes && this.nodes.entries()) {
            this._setInitialCheckedKeys();
        }
    };
    MdbTreeComponent.prototype.toggleExpandAll = function () {
        if (this._expandAll) {
            this.expandAllNodes();
        }
        else if (!this._expandAll) {
            this.closeAllNodes();
        }
    };
    MdbTreeComponent.prototype.expandAllNodes = function () {
        var e_1, _a;
        try {
            for (var _b = __values(this.nodes.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), index = _d[0], node = _d[1];
                var idx = index;
                this.toggle[idx] = true;
                if (node[this.childrenField] && node[this.childrenField].length > 0) {
                    this._expandAllChildren(node, idx);
                }
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_1) throw e_1.error; }
        }
    };
    MdbTreeComponent.prototype.closeAllNodes = function () {
        var e_2, _a;
        try {
            for (var _b = __values(this.nodes.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), index = _d[0], node = _d[1];
                var idx = index;
                this.toggle[idx] = false;
                if (node[this.childrenField] && node[this.childrenField].length > 0) {
                    this._closeAllChildren(node, idx);
                }
            }
        }
        catch (e_2_1) { e_2 = { error: e_2_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_2) throw e_2.error; }
        }
    };
    MdbTreeComponent.prototype._expandAllChildren = function (node, idx) {
        var e_3, _a;
        try {
            for (var _b = __values(node[this.childrenField].entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), childIndex = _d[0], childNode = _d[1];
                var childIdx = idx + '_' + childIndex;
                this.toggle[childIdx] = true;
                if (childNode[this.childrenField] && childNode[this.childrenField].length > 0) {
                    this._expandAllChildren(childNode, childIdx);
                }
            }
        }
        catch (e_3_1) { e_3 = { error: e_3_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_3) throw e_3.error; }
        }
    };
    MdbTreeComponent.prototype._closeAllChildren = function (node, idx) {
        var e_4, _a;
        try {
            for (var _b = __values(node[this.childrenField].entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), childIndex = _d[0], childNode = _d[1];
                var childIdx = idx + '_' + childIndex;
                this.toggle[childIdx] = false;
                if (childNode[this.childrenField] && childNode[this.childrenField].length > 0) {
                    this._closeAllChildren(childNode, childIdx);
                }
            }
        }
        catch (e_4_1) { e_4 = { error: e_4_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_4) throw e_4.error; }
        }
    };
    MdbTreeComponent.prototype._setInitialCheckedKeys = function () {
        var e_5, _a;
        try {
            for (var _b = __values(this.nodes.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), index = _d[0], node = _d[1];
                if (node[this.checkboxesField]) {
                    var idx = index;
                    this.checkedValues.push(idx);
                    if (node[this.childrenField] && node[this.childrenField].length > 0) {
                        this._hasInitialCheckedKeysChildren(node[this.childrenField], idx);
                    }
                }
            }
        }
        catch (e_5_1) { e_5 = { error: e_5_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_5) throw e_5.error; }
        }
    };
    MdbTreeComponent.prototype._hasInitialCheckedKeysChildren = function (childrenNode, i) {
        var e_6, _a;
        try {
            for (var _b = __values(childrenNode.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), childrenIdx = _d[0], node = _d[1];
                var idx = childrenIdx + '_' + i;
                if (node[this.checkboxesField]) {
                    this.checkedValues.push(idx);
                }
                if (node[this.childrenField] && node[this.childrenField].length > 0) {
                    this._hasInitialCheckedKeysChildren(node[this.childrenField], idx);
                }
            }
        }
        catch (e_6_1) { e_6 = { error: e_6_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_6) throw e_6.error; }
        }
    };
    MdbTreeComponent.prototype.toggleByNode = function (i) {
        var e_7, _a;
        try {
            for (var _b = __values(this.nodes.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), index = _d[0], node = _d[1];
                if (node[this.childrenField] && node[this.childrenField].length > 0) {
                    var idx = index;
                    var toggleIdx = i;
                    if (idx === toggleIdx) {
                        this.toggle[idx] = !this.toggle[idx];
                        this._cdRef.markForCheck();
                    }
                    else {
                        this._childrenToggleByNode(node, idx, toggleIdx);
                    }
                }
            }
        }
        catch (e_7_1) { e_7 = { error: e_7_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_7) throw e_7.error; }
        }
    };
    MdbTreeComponent.prototype._childrenToggleByNode = function (node, i, toggleIdx) {
        var e_8, _a;
        try {
            for (var _b = __values(node[this.childrenField].entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), childIndex = _d[0], childNode = _d[1];
                var nodeHasChildren = childNode[this.childrenField] && childNode[this.childrenField].length > 0;
                if (nodeHasChildren) {
                    var idx = i + '_' + childIndex;
                    if (idx === toggleIdx) {
                        this.toggle[idx] = !this.toggle[idx];
                        this._cdRef.markForCheck();
                    }
                    else {
                        this._childrenToggleByNode(childNode, idx, toggleIdx);
                    }
                }
                else {
                    return;
                }
            }
        }
        catch (e_8_1) { e_8 = { error: e_8_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_8) throw e_8.error; }
        }
    };
    MdbTreeComponent.prototype.onKeydownCheckbox = function (e, node, i) {
        // tslint:disable-next-line: deprecation
        if (e.keyCode === SPACE || e.keyCode === ENTER) {
            e.preventDefault();
            this.checkNodes(node);
            this.updateNodesCheckedValues(node, i);
        }
    };
    MdbTreeComponent.prototype.onKeydown = function (e, i) {
        // tslint:disable-next-line: deprecation
        if (e.keyCode === SPACE || e.keyCode === ENTER) {
            e.preventDefault();
            this.toggle[i] = !this.toggle[i];
        }
    };
    MdbTreeComponent.prototype.checkNodes = function (node) {
        var _this = this;
        setTimeout(function () {
            node[_this.checkboxesField] = !node[_this.checkboxesField];
            _this.checked.emit(node);
            _this.nodesChanged.emit(_this.nodes);
        }, 0);
        var nodeHasChildren = node[this.childrenField] && node[this.childrenField].length > 0;
        if (nodeHasChildren) {
            this._checkChildNodes(node[this.childrenField], !node[this.checkboxesField]);
        }
        this._cdRef.markForCheck();
    };
    MdbTreeComponent.prototype._checkChildNodes = function (children, checked) {
        var _this = this;
        children.forEach(function (childNode) {
            if (childNode[_this.checkboxesField] !== undefined) {
                childNode[_this.checkboxesField] = checked;
                var nodeHasChildren = childNode[_this.childrenField] && childNode[_this.childrenField].length > 0;
                if (nodeHasChildren) {
                    _this._checkChildNodes(childNode[_this.childrenField], checked);
                }
            }
        });
    };
    MdbTreeComponent.prototype.updateNodesCheckedValues = function (node, idx) {
        var _this = this;
        setTimeout(function () {
            if (node[_this.checkboxesField] && !_this.checkedValues.includes(idx)) {
                _this.checkedValues.push(idx);
            }
            else if (!node[_this.checkboxesField] && _this.checkedValues.includes(idx)) {
                var removeIndex = _this.checkedValues.findIndex(function (e) { return e === idx; });
                if (removeIndex !== -1) {
                    _this.checkedValues.splice(removeIndex, 1);
                }
            }
            var nodeHasChildren = node[_this.childrenField] && node[_this.childrenField].length > 0;
            if (nodeHasChildren) {
                _this._updateChildNodesCheckedValues(node[_this.childrenField], idx);
            }
            _this.checkedKeys.emit(_this.checkedValues);
        }, 0);
    };
    MdbTreeComponent.prototype._updateChildNodesCheckedValues = function (childrenNode, childrenIdx) {
        var e_9, _a;
        var _loop_1 = function (index, node) {
            var idx = childrenIdx + '_' + index;
            if (node[this_1.checkboxesField] && !this_1.checkedValues.includes(idx)) {
                this_1.checkedValues.push(idx);
            }
            else if (!node[this_1.checkboxesField] && this_1.checkedValues.includes(idx)) {
                var removeIndex = this_1.checkedValues.findIndex(function (e) { return e === idx; });
                if (removeIndex !== -1) {
                    this_1.checkedValues.splice(removeIndex, 1);
                }
            }
            var nodeHasChildren = node[this_1.childrenField] && node[this_1.childrenField].length > 0;
            if (nodeHasChildren) {
                this_1._updateChildNodesCheckedValues(node[this_1.childrenField], idx);
            }
        };
        var this_1 = this;
        try {
            for (var _b = __values(childrenNode.entries()), _c = _b.next(); !_c.done; _c = _b.next()) {
                var _d = __read(_c.value, 2), index = _d[0], node = _d[1];
                _loop_1(index, node);
            }
        }
        catch (e_9_1) { e_9 = { error: e_9_1 }; }
        finally {
            try {
                if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
            }
            finally { if (e_9) throw e_9.error; }
        }
    };
    MdbTreeComponent.ctorParameters = function () { return [
        { type: ChangeDetectorRef }
    ]; };
    __decorate([
        HostBinding('class.mdb-tree'),
        Output(),
        __metadata("design:type", Object)
    ], MdbTreeComponent.prototype, "checked", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], MdbTreeComponent.prototype, "checkedKeys", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], MdbTreeComponent.prototype, "nodesChanged", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], MdbTreeComponent.prototype, "nodes", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], MdbTreeComponent.prototype, "textField", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], MdbTreeComponent.prototype, "childrenField", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], MdbTreeComponent.prototype, "checkboxesField", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean),
        __metadata("design:paramtypes", [Boolean])
    ], MdbTreeComponent.prototype, "expandAll", null);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], MdbTreeComponent.prototype, "checkboxes", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], MdbTreeComponent.prototype, "toggleOnTitleClick", void 0);
    MdbTreeComponent = __decorate([
        Component({
            // tslint:disable-next-line: component-selector
            selector: 'mdb-tree',
            template: "<!-- child nodes -->\n<ng-template #tree let-nodeChildren let-idx=\"idx\">\n  <ul class=\"mdb-tree-list\">\n    <li *ngFor=\"let node of nodeChildren; let n = index\" class=\"mdb-tree-list-node\">\n      <div class=\"mdb-tree-container\">\n        <div *ngIf=\"node[childrenField] && node[childrenField].length > 0; else emptyIcon\">\n          <span class=\"mdb-tree-icon-container\">\n            <i\n              tabindex=\"1\"\n              aria-hidden=\"true\"\n              [ngClass]=\"\n                toggle[idx + '_' + n] ? 'mdb-tree-rotate-icon-open' : 'mdb-tree-rotate-icon-closed'\n              \"\n              (keydown)=\"onKeydown($event, idx + '_' + n)\"\n              (click)=\"toggle[idx + '_' + n] = !toggle[idx + '_' + n]\"\n              class=\"mdb-tree-indicator \"\n            ></i>\n          </span>\n        </div>\n        <ng-template #emptyIcon\n          ><span class=\"mdb-tree-icon-container\"\n            ><i class=\"mdb-tree-empty-icon\" style=\"display: block\" aria-hidden=\"true\"></i\n          ></span>\n        </ng-template>\n        <div\n          class=\"mdb-tree-checkbox-container\"\n          *ngIf=\"checkboxes && node[checkboxesField] !== undefined\"\n        >\n          <mdb-checkbox\n            class=\"checkbox-filled\"\n            [filledIn]=\"true\"\n            [tabIndex]=\"1\"\n            [attr.id]=\"node[textField]\"\n            (keydown)=\"onKeydownCheckbox($event, node, idx + '_' + n)\"\n            (click)=\"checkNodes(node); updateNodesCheckedValues(node, idx + '_' + n)\"\n            [checked]=\"node[checkboxesField]\"\n          ></mdb-checkbox>\n        </div>\n        <div *ngIf=\"checkboxes && node[checkboxesField] === undefined\">\n          <div class=\"mdb-tree-checkbox-null-container\"></div>\n        </div>\n\n        <div\n          *ngIf=\"toggleOnTitleClick\"\n          class=\"mdb-tree-text-field\"\n          [ngStyle]=\"{\n            cursor: node[childrenField] && node[childrenField].length > 0 ? 'pointer' : 'default'\n          }\"\n          (click)=\"toggle[idx + '_' + n] = !toggle[idx + '_' + n]\"\n        >\n          {{ node[textField] }}\n        </div>\n\n        <div *ngIf=\"!toggleOnTitleClick\" class=\"mdb-tree-text-field mdb-tree-text-ellipsis\">\n          {{ node[textField] }}\n        </div>\n      </div>\n      <div *ngIf=\"node[childrenField] && toggle[idx + '_' + n]\">\n        <ng-container\n          *ngTemplateOutlet=\"tree; context: { $implicit: node[childrenField], idx: idx + '_' + n }\"\n        ></ng-container>\n      </div>\n    </li>\n  </ul>\n</ng-template>\n<!-- first nodes -->\n<ul class=\"mdb-tree-list\">\n  <li *ngFor=\"let node of nodes; let i = index\" class=\"mdb-tree-list-node\">\n    <div class=\"mdb-tree-container\">\n      <div *ngIf=\"node[childrenField] && node[childrenField].length > 0; else emptyIcon\">\n        <span class=\"mdb-tree-icon-container\">\n          <i\n            tabindex=\"1\"\n            aria-hidden=\"true\"\n            [ngClass]=\"toggle[i] ? 'mdb-tree-rotate-icon-open' : 'mdb-tree-rotate-icon-closed'\"\n            (keydown)=\"onKeydown($event, i)\"\n            (click)=\"toggle[i] = !toggle[i]\"\n            class=\"mdb-tree-indicator\"\n          ></i>\n        </span>\n      </div>\n      <ng-template #emptyIcon\n        ><span class=\"mdb-tree-icon-container\"\n          ><i class=\"mdb-tree-empty-icon\" style=\"display: block\" aria-hidden=\"true\"></i\n        ></span>\n      </ng-template>\n      <div\n        class=\"mdb-tree-checkbox-container\"\n        *ngIf=\"checkboxes && node[checkboxesField] !== undefined\"\n      >\n        <mdb-checkbox\n          class=\"checkbox-filled\"\n          [checked]=\"node[checkboxesField]\"\n          [filledIn]=\"true\"\n          [tabIndex]=\"1\"\n          [attr.id]=\"node[textField]\"\n          (keydown)=\"onKeydownCheckbox($event, node, i)\"\n          (click)=\"checkNodes(node); updateNodesCheckedValues(node, i)\"\n        ></mdb-checkbox>\n      </div>\n      <div *ngIf=\"checkboxes && node[checkboxesField] === undefined\">\n        <div class=\"mdb-tree-checkbox-null-container\"></div>\n      </div>\n\n      <div\n        *ngIf=\"toggleOnTitleClick\"\n        class=\"mdb-tree-text-field\"\n        [ngStyle]=\"{\n          cursor: node[childrenField] && node[childrenField].length > 0 ? 'pointer' : 'default'\n        }\"\n        (click)=\"toggle[i] = !toggle[i]\"\n      >\n        {{ node[textField] }}\n      </div>\n\n      <div *ngIf=\"!toggleOnTitleClick\" class=\"mdb-tree-text-field mdb-tree-text-ellipsis\">\n        {{ node[textField] }}\n      </div>\n    </div>\n    <div *ngIf=\"node[childrenField] && toggle[i]\">\n      <ng-container\n        *ngTemplateOutlet=\"tree; context: { $implicit: node[childrenField], idx: i }\"\n      ></ng-container>\n    </div>\n  </li>\n</ul>\n",
            encapsulation: ViewEncapsulation.None,
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [".mdb-tree-list{list-style-type:none;margin:0;padding:0}.mdb-tree-list-node{list-style-type:none;margin:.8rem .8rem .8rem .95rem}.mdb-tree-container{display:flex;min-width:230px}.mdb-tree-icon-container{display:inline-block;width:2rem;height:auto}.mdb-tree-empty-icon{cursor:default}.mdb-tree-text-field{margin-top:.15rem;max-width:90%;text-overflow:ellipsis;overflow:hidden;white-space:nowrap}.mdb-tree-checkbox-null-container{min-width:2.2rem}.mdb-tree-indicator{position:relative;right:0;transform-origin:50% 79%;display:inline-block;margin-right:0;margin-top:.025rem;cursor:pointer;font-size:1.3rem}.mdb-tree-indicator::after{content:\"\";display:block;border-style:solid;padding:5px;margin-top:.15rem;border-width:0 3px 3px 0;font-size:1.3rem;transform:rotate(45deg)}.mdb-tree-indicator:focus{color:#4285f4;outline:0}.mdb-tree-rotate-icon-open{transform:rotate(0)}.mdb-tree-rotate-icon-closed{transform:rotate(270deg)}.mdb-tree-checkbox-container{margin-top:.25rem}.mdb-tree-checkbox-container mdb-checkbox.checkbox-filled [type=checkbox][class*=filled-in]:checked+label:after{border-color:#4285f4;background-color:#4285f4}"]
        }),
        __metadata("design:paramtypes", [ChangeDetectorRef])
    ], MdbTreeComponent);
    return MdbTreeComponent;
}());
export { MdbTreeComponent };
