// ng-uikit-pro-standard/esm5/lib/pro/lightbox/light-box.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { ImageModalComponent } from './image-popup';
import * as ɵngcc0 from '@angular/core';
var LightBoxModule = /** @class */ (function () {
    function LightBoxModule() {
    }
LightBoxModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LightBoxModule });
LightBoxModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function LightBoxModule_Factory(t) { return new (t || LightBoxModule)(); }, imports: [[CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LightBoxModule, { declarations: function () { return [ImageModalComponent]; }, imports: function () { return [CommonModule, FormsModule]; }, exports: function () { return [ImageModalComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LightBoxModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule],
                declarations: [ImageModalComponent],
                exports: [ImageModalComponent]
            }]
    }], function () { return []; }, null); })();
    return LightBoxModule;
}());
export { LightBoxModule };
