// ng-uikit-pro-standard/esm5/lib/pro/sidenav/sidenav.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SidenavComponent } from './sidenav.component';
import * as ɵngcc0 from '@angular/core';
var SidenavModule = /** @class */ (function () {
    function SidenavModule() {
    }
SidenavModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SidenavModule });
SidenavModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SidenavModule_Factory(t) { return new (t || SidenavModule)(); }, imports: [[
            CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SidenavModule, { declarations: function () { return [SidenavComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SidenavComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidenavModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    SidenavComponent,
                ],
                exports: [
                    SidenavComponent
                ],
                imports: [
                    CommonModule,
                ]
            }]
    }], function () { return []; }, null); })();
    return SidenavModule;
}());
export { SidenavModule };
