// ng-uikit-pro-standard/esm2015/lib/pro/auto-completer/components/mdb-option.component.js
import { Component, ElementRef, Input, HostListener, InjectionToken, Optional, Inject, } from '@angular/core';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../directives/mdb-auto-completer-option.directive';
import * as ɵngcc2 from '@angular/common';

const _c0 = function (a0) { return { "height.px": a0 }; };
const _c1 = ["*"];
export const MDB_OPTION_PARENT = new InjectionToken('MDB_OPTION_PARENT');
export class MdbOptionComponent {
    constructor(el, _parent) {
        this.el = el;
        this._parent = _parent;
        this.clicked = false;
        this.clickSource = new Subject();
        this.click$ = this.clickSource.asObservable();
        this.clicked = false;
    }
    get optionHeight() {
        return this._optionHeight;
    }
    onClick() {
        this.clickSource.next(this);
    }
    get label() {
        return this.el.nativeElement.textContent;
    }
    ngOnInit() {
        if (this._parent.visibleOptions && this._parent.optionHeight) {
            this._optionHeight = this._parent.optionHeight;
        }
    }
}
MdbOptionComponent.ɵfac = function MdbOptionComponent_Factory(t) { return new (t || MdbOptionComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(MDB_OPTION_PARENT, 8)); };
MdbOptionComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MdbOptionComponent, selectors: [["mdb-option"]], hostBindings: function MdbOptionComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function MdbOptionComponent_click_HostBindingHandler() { return ctx.onClick(); });
    } }, inputs: { value: "value", disabled: "disabled" }, ngContentSelectors: _c1, decls: 2, vars: 3, consts: [["tabindex", "0", "mdbAutoCompleterOption", "", 1, "completer-row", 3, "ngStyle"]], template: function MdbOptionComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(1, _c0, ctx._optionHeight ? ctx._optionHeight : "auto"));
    } }, directives: [ɵngcc1.MdbAutoCompleterOptionDirective, ɵngcc2.NgStyle], encapsulation: 2 });
MdbOptionComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [MDB_OPTION_PARENT,] }] }
];
MdbOptionComponent.propDecorators = {
    value: [{ type: Input }],
    disabled: [{ type: Input }],
    onClick: [{ type: HostListener, args: ['click',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbOptionComponent, [{
        type: Component,
        args: [{
                selector: 'mdb-option',
                template: "<div\n  tabindex=\"0\"\n  class=\"completer-row\"\n  [ngStyle]=\"{\n    'height.px': _optionHeight ? _optionHeight : 'auto'\n  }\"\n  mdbAutoCompleterOption\n>\n  <ng-content></ng-content>\n</div>\n"
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: undefined, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [MDB_OPTION_PARENT]
            }] }]; }, { onClick: [{
            type: HostListener,
            args: ['click']
        }], value: [{
            type: Input
        }], disabled: [{
            type: Input
        }] }); })();
