// ng-uikit-pro-standard/esm5/lib/pro/tabs-pills/tabset.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgTranscludeDirective } from './transclude.directive';
import { TabHeadingDirective } from './tab-heading.directive';
import { TabDirective } from './tab.directive';
import { TabsetComponent } from './tabset.component';
import { TabsetConfig } from './tabset.config';
var TabsModule = /** @class */ (function () {
    function TabsModule() {
    }
    TabsModule_1 = TabsModule;
    TabsModule.forRoot = function () {
        return {
            ngModule: TabsModule_1,
            providers: [TabsetConfig],
        };
    };
    var TabsModule_1;
    TabsModule = TabsModule_1 = __decorate([
        NgModule({
            imports: [CommonModule],
            declarations: [NgTranscludeDirective, TabDirective, TabsetComponent, TabHeadingDirective],
            exports: [TabDirective, TabsetComponent, TabHeadingDirective, NgTranscludeDirective],
        })
    ], TabsModule);
    return TabsModule;
}());
export { TabsModule };
