// ng-uikit-pro-standard/esm5/lib/pro/timepicker/timepicker-toggle.component.js
import { __decorate, __metadata } from "tslib";
import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input, HostListener, } from '@angular/core';
import { MdbTimePickerComponent } from './timepicker.component';
import * as ɵngcc0 from '@angular/core';
var MdbTimepickerToggleComponent = /** @class */ (function () {
    function MdbTimepickerToggleComponent() {
    }
    MdbTimepickerToggleComponent.prototype.handleClick = function () {
        this.mdbTimePickerToggle.open();
    };
    MdbTimepickerToggleComponent.prototype.ngOnInit = function () { };
    __decorate([
        Input(),
        __metadata("design:type", MdbTimePickerComponent)
    ], MdbTimepickerToggleComponent.prototype, "mdbTimePickerToggle", void 0);
    __decorate([
        HostListener('click'),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], MdbTimepickerToggleComponent.prototype, "handleClick", null);
    MdbTimepickerToggleComponent = __decorate([ __metadata("design:paramtypes", [])
    ], MdbTimepickerToggleComponent);
MdbTimepickerToggleComponent.ɵfac = function MdbTimepickerToggleComponent_Factory(t) { return new (t || MdbTimepickerToggleComponent)(); };
MdbTimepickerToggleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MdbTimepickerToggleComponent, selectors: [["mdb-timepicker-toggle"]], hostBindings: function MdbTimepickerToggleComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function MdbTimepickerToggleComponent_click_HostBindingHandler() { return ctx.handleClick(); });
    } }, inputs: { mdbTimePickerToggle: "mdbTimePickerToggle" }, decls: 3, vars: 0, consts: [[1, "mdb-timepicker-toggle"], ["aria-hidden", "true", "focusable", "false", "data-prefix", "fas", "data-icon", "clock", "role", "img", "xmlns", "http://www.w3.org/2000/svg", "viewBox", "0 0 512 512", 1, "svg-inline--fa", "fa-clock", "fa-w-16"], ["fill", "currentColor", "d", "M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"]], template: function MdbTimepickerToggleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "button", 0);
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelementStart(1, "svg", 1);
        ɵngcc0.ɵɵelement(2, "path", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } }, styles: [".mdb-timepicker-toggle{position:absolute;right:0;top:50%;transform:translateY(-50%);border-color:transparent;background-color:transparent}.mdb-timepicker-toggle svg{height:1em}"], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbTimepickerToggleComponent, [{
        type: Component,
        args: [{
                template: "<button class=\"mdb-timepicker-toggle\">\n  <svg\n    aria-hidden=\"true\"\n    focusable=\"false\"\n    data-prefix=\"fas\"\n    data-icon=\"clock\"\n    class=\"svg-inline--fa fa-clock fa-w-16\"\n    role=\"img\"\n    xmlns=\"http://www.w3.org/2000/svg\"\n    viewBox=\"0 0 512 512\"\n  >\n    <path\n      fill=\"currentColor\"\n      d=\"M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z\"\n    ></path>\n  </svg>\n</button>\n",
                selector: 'mdb-timepicker-toggle',
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [".mdb-timepicker-toggle{position:absolute;right:0;top:50%;transform:translateY(-50%);border-color:transparent;background-color:transparent}.mdb-timepicker-toggle svg{height:1em}"]
            }]
    }], function () { return []; }, { handleClick: [{
            type: HostListener,
            args: ['click']
        }], mdbTimePickerToggle: [{
            type: Input
        }] }); })();
    return MdbTimepickerToggleComponent;
}());
export { MdbTimepickerToggleComponent };
