// ng-uikit-pro-standard/esm5/lib/pro/timepicker/timepicker.content.js
import { __assign, __decorate, __metadata, __param } from "tslib";
import { DOCUMENT } from '@angular/common';
import { AfterViewInit, ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, Inject, NgZone, OnInit, Renderer2, ViewChild, ViewEncapsulation, } from '@angular/core';
import { FocusMonitor } from '@angular/cdk/a11y';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/cdk/a11y';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../free/waves/waves-effect.directive';

var _c0 = ["plate"];
var _c1 = ["hand"];
var _c2 = ["fg"];
var _c3 = ["bg"];
var _c4 = ["focus"];
var _c5 = ["digitalMinute"];
var _c6 = function (a0) { return { active: a0 }; };
function MdbTimePickerContentComponent_div_10_Template(rf, ctx) { if (rf & 1) {
    var _r17 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 34);
    ɵngcc0.ɵɵelementStart(1, "span", 35);
    ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_div_10_Template_span_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r17); var ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16._setAmPm("AM"); })("keydown.enter", function MdbTimePickerContentComponent_div_10_Template_span_keydown_enter_1_listener() { ɵngcc0.ɵɵrestoreView(_r17); var ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18._setAmPm("AM"); });
    ɵngcc0.ɵɵtext(2, "AM");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span", 35);
    ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_div_10_Template_span_click_3_listener() { ɵngcc0.ɵɵrestoreView(_r17); var ctx_r19 = ɵngcc0.ɵɵnextContext(); return ctx_r19._setAmPm("PM"); })("keydown.enter", function MdbTimePickerContentComponent_div_10_Template_span_keydown_enter_3_listener() { ɵngcc0.ɵɵrestoreView(_r17); var ctx_r20 = ɵngcc0.ɵɵnextContext(); return ctx_r20._setAmPm("PM"); });
    ɵngcc0.ɵɵtext(4, "PM");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(2, _c6, ctx_r2._selectedTime.ampm == "AM"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c6, ctx_r2._selectedTime.ampm == "PM"));
} }
var _c7 = function (a0) { return { disabled: a0 }; };
var _c8 = function (a0, a1) { return { left: a0, top: a1 }; };
function MdbTimePickerContentComponent_div_29_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 36);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var tick_r21 = ctx.$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("id", tick_r21.hour);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c7, tick_r21.disabled))("ngStyle", ɵngcc0.ɵɵpureFunction2(6, _c8, tick_r21.left + "px", tick_r21.top + "px"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", tick_r21.hour, " ");
} }
function MdbTimePickerContentComponent_div_32_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 37);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var tick_r22 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(3, _c7, tick_r22.disabled))("ngStyle", ɵngcc0.ɵɵpureFunction2(5, _c8, tick_r22.left + "px", tick_r22.top + "px"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", tick_r22.min, " ");
} }
function MdbTimePickerContentComponent_button_34_Template(rf, ctx) { if (rf & 1) {
    var _r24 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 38);
    ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_button_34_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r24); var ctx_r23 = ɵngcc0.ɵɵnextContext(); return ctx_r23._clearBtnClicked(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r14 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r14.clearButton, " ");
} }
function MdbTimePickerContentComponent_button_35_Template(rf, ctx) { if (rf & 1) {
    var _r26 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 39);
    ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_button_35_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r26); var ctx_r25 = ɵngcc0.ɵɵnextContext(); return ctx_r25._closeBtnClicked(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r15 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r15.closeButton, " ");
} }
var _c9 = function (a0, a1) { return { "active": a0, "disabled": a1 }; };
var _c10 = function (a0) { return { "mdb-timepicker-dial-out": a0 }; };
var _c11 = function (a0) { return { visibility: a0 }; };
var MdbTimePickerContentComponent = /** @class */ (function () {
    function MdbTimePickerContentComponent(_cdRef, _ngZone, focusMonitor, elem, renderer, _document) {
        var _this = this;
        this._cdRef = _cdRef;
        this._ngZone = _ngZone;
        this.focusMonitor = focusMonitor;
        this.elem = elem;
        this.renderer = renderer;
        this._document = _document;
        this._disabledHours = [];
        this._disabledMinutes = [];
        this._isMouseDown = false;
        this._hoursTicks = [];
        this._minuteDigitalDisabled = false;
        this._minutesTicks = [];
        this._okButtonDisabled = false;
        this._showHours = true;
        this._radius = {
            dial: 135,
            inner: 80,
            outer: 110,
            tick: 20,
        };
        this._denominator = {
            1: 30,
            5: 6,
            10: 3,
            15: 2,
            20: 1.5,
        };
        this.touchSupported = 'ontouchstart' in window;
        this._setOkBtnDisabled = function () {
            var hour = Number(_this._to24(_this._selectedTime).h);
            _this._okButtonDisabled = _this._disabledHours[hour];
            if (!_this._okButtonDisabled) {
                if (_this._min &&
                    _this._selectedTime.h === _this._min.h &&
                    _this._selectedTime.ampm === _this._min.ampm) {
                    _this._okButtonDisabled = _this._selectedTime.m < _this._min.m;
                }
                if (_this._max &&
                    _this._selectedTime.h === _this._max.h &&
                    _this._selectedTime.ampm === _this._max.ampm) {
                    _this._okButtonDisabled = _this._selectedTime.m > _this._max.m;
                }
            }
        };
    }
    MdbTimePickerContentComponent.prototype.ngOnInit = function () {
        this._max = this.max;
        this._min = this.min;
        this._selectedTime = this.value;
        var ampm = this._selectedTime.ampm;
        // Add disabled hours to array for PM and AM Hours
        if (this.twelveHour) {
            this._selectedTime.ampm = ampm === 'PM' ? 'AM' : 'PM';
            this._generateTick();
            this._selectedTime.ampm = this._selectedTime.ampm === 'PM' ? 'AM' : 'PM';
        }
        this._generateTick();
        this._setOkBtnDisabled();
        this._setMinuteDigitalDisabled();
    };
    MdbTimePickerContentComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        ['mousedown', 'mouseup', 'touchend', 'touchstart'].forEach(function (event) {
            _this.renderer.listen(_this.plate.nativeElement, event, function (ev) {
                if (event === 'mousedown' || event === 'touchstart') {
                    _this._mousedown(ev, false);
                }
            });
        });
        this._checkDraw();
        setTimeout(function () {
            _this.focusMonitor.focusVia(_this.focus, 'keyboard');
        }, 0);
    };
    MdbTimePickerContentComponent.prototype._checkDraw = function () {
        var _a = this._selectedTime, h = _a.h, m = _a.m;
        var value = this._showHours ? parseInt(h, 0) : parseInt(m, 0);
        var unit = Math.PI / (this._showHours ? 6 : 30), radian = value * unit, radius = this._showHours && value > 0 && value < 13 ? this._radius.inner : this._radius.outer, xd = Math.sin(radian) * radius, yd = -Math.cos(radian) * radius;
        this.setClockHand(xd, yd);
    };
    MdbTimePickerContentComponent.prototype._mousedown = function (e, space) {
        var _this = this;
        this._isMouseDown = true;
        var offset = this.plate.nativeElement.getBoundingClientRect(), isTouch = /^touch/.test(e.type), x0 = offset.left + this._radius.dial, y0 = offset.top + this._radius.dial, dx = (isTouch ? e.touches[0] : e).clientX - x0, dy = (isTouch ? e.touches[0] : e).clientY - y0, z = Math.sqrt(dx * dx + dy * dy);
        var moved = false;
        if (space &&
            (z < this._radius.outer - this._radius.tick || z > this._radius.outer + this._radius.tick)) {
            return;
        }
        e.preventDefault();
        e.stopPropagation();
        if (this._showHours) {
            this.setClockHand(dx, dy);
        }
        else {
            this.setClockHand(dx, dy, this.rounding);
        }
        var mousemoveEventMethod = function (event) {
            if (!_this.touchSupported) {
                event.preventDefault();
            }
            event.stopPropagation();
            // tslint:disable-next-line:no-shadowed-variable
            var isTouch = /^touch/.test(event.type), x = (isTouch ? event.touches[0] : event).clientX - x0, y = (isTouch ? event.touches[0] : event).clientY - y0;
            if (!moved && x === dx && y === dy) {
                return;
            }
            moved = true;
            _this._ngZone.run(function () {
                _this.setClockHand(x, y, _this.rounding);
            });
        };
        var mouseupEventMethod = function (event) {
            _this._document.removeEventListener('mousemove', mousemoveEventMethod);
            if (_this.touchSupported) {
                _this._document.removeEventListener('touchmove', mousemoveEventMethod);
            }
            if (!_this.touchSupported) {
                event.preventDefault();
            }
            var x = event.clientX - x0, y = event.clientY - y0;
            if ((space || moved) && x === dx && y === dy) {
                _this.setClockHand(x, y);
            }
            _this._ngZone.run(function () {
                if (_this.autoClose && !_this._showHours) {
                    _this._okBtnClicked();
                }
            });
            _this._showMinutesClock();
            _this.digitalMinute.nativeElement.focus();
            _this._isMouseDown = false;
            _this._document.removeEventListener('mouseup', mouseupEventMethod);
            if (_this.touchSupported) {
                _this._document.removeEventListener('touchend', mouseupEventMethod);
            }
            _this.picker._emitTimeChangeEvent(_this._selectedTime);
        };
        this._document.addEventListener('mousemove', mousemoveEventMethod);
        if (this.touchSupported) {
            this._document.addEventListener('touchmove', mousemoveEventMethod);
        }
        this._document.addEventListener('mouseup', mouseupEventMethod);
        if (this.touchSupported) {
            this._document.addEventListener('touchend', mouseupEventMethod);
        }
    };
    MdbTimePickerContentComponent.prototype._closeBtnClicked = function () {
        // todo this.isOpen = false;
        var _a = this._selectedTime, ampm = _a.ampm, h = _a.h, m = _a.m;
        this._returnHours = this.twelveHour ? h + ":" + m + ampm : h + ":" + m + ampm;
        this.picker.close(false);
    };
    MdbTimePickerContentComponent.prototype._clearBtnClicked = function () {
        this._setAmPm('AM');
        this._setHour(12);
        this._setMinute(0);
        this._generateTick();
        this._showHoursClock();
        this.picker._setValue('');
        this.picker._selectionChange$.next('');
    };
    MdbTimePickerContentComponent.prototype._okBtnClicked = function () {
        if (!this._okButtonDisabled) {
            var _a = this._selectedTime, ampm = _a.ampm, h = _a.h, m = _a.m;
            this._returnHours = this.twelveHour ? h + ":" + m + ampm : h + ":" + m + ampm;
            this.picker._setValue(this._returnHours);
            this.picker._emitTimeDoneEvent(this._selectedTime);
            this.picker.onChangeCb(this._returnHours);
            this.picker.close(true);
        }
    };
    MdbTimePickerContentComponent.prototype._arrowChangeHour = function (key) {
        var _a = this._to24(this._selectedTime), h = _a.h, ampm = _a.ampm;
        var selectedHour = Number(h);
        var availableHours = [];
        this._disabledHours.map(function (hour, index) { return !hour && availableHours.push(index); });
        var toChange;
        var value = key === 'ArrowUp'
            ? availableHours.indexOf(selectedHour) + 1
            : availableHours.indexOf(selectedHour) - 1;
        value = value < 0 ? availableHours.length - 1 : value;
        value = value > availableHours.length - 1 ? 0 : value;
        toChange = availableHours[value];
        if (this.twelveHour) {
            if (toChange >= 12) {
                toChange = toChange - 12;
                if (ampm === 'AM') {
                    this._setAmPm('PM');
                }
            }
            else if (toChange <= 0 || toChange < 12) {
                if (ampm === 'PM') {
                    this._setAmPm('AM');
                }
            }
        }
        this._showHoursClock();
        this._setHour(toChange);
        this._checkDraw();
    };
    MdbTimePickerContentComponent.prototype._arrowChangeMinute = function (key) {
        if (!this._minuteDigitalDisabled) {
            if (this._showHours) {
                this._showMinutesClock();
            }
            var m = this._selectedTime.m;
            var availableMinutes_1 = [];
            this._generateMinutesDisable();
            this._disabledMinutes.map(function (disabled, i) {
                if (!disabled) {
                    availableMinutes_1.push(i);
                }
            });
            var toChange = void 0;
            var value = key === 'ArrowUp'
                ? availableMinutes_1.indexOf(Number(m)) + 1
                : availableMinutes_1.indexOf(Number(m)) - 1;
            value = value < 0 ? availableMinutes_1.length - 1 : value;
            value = value > availableMinutes_1.length - 1 ? 0 : value;
            toChange = availableMinutes_1[value];
            this._setMinute(toChange);
            this._checkDraw();
        }
    };
    MdbTimePickerContentComponent.prototype._generateMinutesDisable = function () {
        for (var i = 0; i < 60; i++) {
            var disableByRounding = this.rounding > 1 && i % this.rounding !== 0;
            var disabled = this._rangeMinute(i, 'min') || this._rangeMinute(i, 'max') || disableByRounding;
            this._disabledMinutes[i] = disabled;
        }
    };
    MdbTimePickerContentComponent.prototype._setHour = function (hour) {
        if (Number(this._selectedTime.h) !== hour) {
            if (this.twelveHour) {
                hour = hour <= 0 ? 12 : hour;
                hour = hour > 12 ? 1 : hour;
            }
            else {
                hour = hour >= 24 ? 0 : hour;
                hour = hour <= -1 ? 23 : hour;
            }
            this._selectedTime.h = hour >= 10 ? "" + hour : "0" + hour;
            this._setMinuteDigitalDisabled();
        }
    };
    MdbTimePickerContentComponent.prototype._setMinute = function (min) {
        if (Number(this._selectedTime.m) !== min) {
            min = min >= 60 ? 0 : min;
            min = min <= -1 ? 59 : min;
            this._selectedTime.m = min >= 10 ? "" + min : "0" + min;
            this._setOkBtnDisabled();
        }
    };
    MdbTimePickerContentComponent.prototype._setAmPm = function (ampm) {
        this._selectedTime.ampm = ampm;
        this._generateTick();
        this._setOkBtnDisabled();
        this._setMinuteDigitalDisabled();
        this._checkDraw();
        this.picker._emitTimeChangeEvent(this._selectedTime);
    };
    MdbTimePickerContentComponent.prototype._showHoursClock = function () {
        this._generateTick();
        this._showHours = true;
        this._setOkBtnDisabled();
        this._checkDraw();
    };
    MdbTimePickerContentComponent.prototype._showMinutesClock = function () {
        if (!this._minuteDigitalDisabled) {
            this._showHours = false;
            this._generateTick();
            this._setOkBtnDisabled();
            this._generateMinutesDisable();
            if (this._disabledMinutes[Number(this._selectedTime.m)] === true) {
                this._setMinute(this._disabledMinutes.indexOf(false));
            }
            this._checkDraw();
        }
    };
    MdbTimePickerContentComponent.prototype._generateTick = function () {
        if (this.twelveHour) {
            this._hoursTicks = [];
            for (var i = 1; i < 13; i++) {
                var radian = (i / 6) * Math.PI;
                var tick = {
                    hour: i.toString(),
                    left: this._radius.dial + Math.sin(radian) * this._radius.outer - this._radius.tick,
                    top: this._radius.dial - Math.cos(radian) * this._radius.outer - this._radius.tick,
                    disabled: this._rangeHour(i, 'min') || this._rangeHour(i, 'max'),
                };
                this._hoursTicks.push(tick);
            }
        }
        else {
            this._hoursTicks = [];
            for (var i = 0; i < 24; i++) {
                var radian = (i / 6) * Math.PI;
                var inner = i > 0 && i < 13;
                var radius = inner ? this._radius.inner : this._radius.outer;
                var hour = i === 0 ? '0' + i.toString() : i.toString();
                var tick = {
                    hour: hour,
                    left: this._radius.dial + Math.sin(radian) * radius - this._radius.tick,
                    top: this._radius.dial - Math.cos(radian) * radius - this._radius.tick,
                    disabled: this._rangeHour(i, 'min') || this._rangeHour(i, 'max'),
                };
                this._hoursTicks.push(tick);
            }
        }
        this._minutesTicks = [];
        for (var i = 0; i < 60; i += 5) {
            var radian = (i / 30) * Math.PI;
            var disableByRounding = this.rounding > 1 && i % this.rounding !== 0;
            var min = i < 10 ? '0' + i.toString() : i.toString();
            var tick = {
                min: min,
                left: this._radius.dial + Math.sin(radian) * this._radius.outer - this._radius.tick,
                top: this._radius.dial - Math.cos(radian) * this._radius.outer - this._radius.tick,
                disabled: this._rangeMinute(i, 'min') || this._rangeMinute(i, 'max') || disableByRounding,
            };
            this._minutesTicks.push(tick);
        }
    };
    MdbTimePickerContentComponent.prototype.setClockHand = function (x, y, roundBy) {
        var radian = Math.atan2(x, -y);
        var isHours = this._showHours;
        var unit = Math.PI / (isHours ? 6 : roundBy ? this._denominator[roundBy] : 30);
        var z = Math.sqrt(x * x + y * y);
        var inner = isHours && z < (this._radius.outer + this._radius.inner) / 2;
        var value = this._showHours
            ? parseInt(this._selectedTime.h, 0)
            : parseInt(this._selectedTime.m, 0);
        var radius = inner && !this.twelveHour ? this._radius.inner : this._radius.outer;
        if (radian < 0) {
            radian = Math.PI * 2 + radian;
        }
        value = Math.round(radian / unit);
        radian = value * unit;
        if (this.twelveHour && isHours) {
            if (value === 0) {
                value = 12;
            }
            if (this._isMouseDown) {
                if (isHours && (this._rangeHour(value, 'min') || this._rangeHour(value, 'max'))) {
                    return;
                }
            }
        }
        else if (!this.twelveHour && isHours) {
            value = !inner ? value + 12 : value;
            value = value === 24 ? 0 : value;
            value = inner && value === 0 ? 12 : value;
            value = !inner && value === 12 ? 0 : value;
            if (this._isMouseDown) {
                if (isHours && (this._rangeHour(value, 'min') || this._rangeHour(value, 'max'))) {
                    return;
                }
            }
        }
        else {
            if (roundBy) {
                value *= roundBy;
            }
            if (value === 60) {
                value = 0;
            }
        }
        if (isHours) {
            this.fg.nativeElement.setAttribute('class', 'mdb-timepicker-canvas-fg');
        }
        else {
            if (this._rangeMinute(value, 'min') || this._rangeMinute(value, 'max')) {
                this._cdRef.markForCheck();
                return;
            }
            if (value % 5 === 0) {
                this.fg.nativeElement.setAttribute('class', 'mdb-timepicker-canvas-fg');
            }
            else {
                this.fg.nativeElement.setAttribute('class', 'mdb-timepicker-canvas-fg active');
            }
        }
        var cx1 = Math.sin(radian) * (radius - this._radius.tick), cy1 = -Math.cos(radian) * (radius - this._radius.tick), cx2 = Math.sin(radian) * radius, cy2 = -Math.cos(radian) * radius;
        this.hand.nativeElement.setAttribute('x2', cx1);
        this.hand.nativeElement.setAttribute('y2', cy1);
        this.bg.nativeElement.setAttribute('cx', cx2);
        this.bg.nativeElement.setAttribute('cy', cy2);
        this.fg.nativeElement.setAttribute('cx', cx2);
        this.fg.nativeElement.setAttribute('cy', cy2);
        if (this._showHours) {
            if (value !== Number(this._selectedTime.h)) {
                this._setHour(value);
                this._setMinuteDigitalDisabled();
            }
        }
        else {
            if (value !== Number(this._selectedTime.m)) {
                this._setMinute(value);
            }
        }
        this._cdRef.markForCheck();
    };
    MdbTimePickerContentComponent.prototype._to24 = function (time) {
        var hour = time.ampm === 'PM' ? Number(time.h) + 12 : Number(time.h);
        hour = hour === 12 ? 0 : hour;
        hour = hour === 24 ? 12 : hour;
        return __assign(__assign({}, time), { h: "" + hour });
    };
    MdbTimePickerContentComponent.prototype._rangeHour = function (index, range) {
        var status = false;
        var i = Number(this._to24(__assign(__assign({}, this._selectedTime), { h: "" + index })).h);
        if (!this.twelveHour) {
            var minH = this.min && Number(this._min.h);
            var maxH = this.max && Number(this._max.h);
            if (range === 'min' && this.min) {
                status = index < minH;
                if (status && this._max && this._max.h < this._min.h) {
                    status = false;
                }
            }
            else if (range === 'max' && this.max) {
                status = index > maxH;
                if (status && this._min && this._min.h > this._max.h && minH <= index) {
                    status = false;
                }
            }
        }
        else {
            var min = this._min && Number(this._to24(this._min).h);
            var max = this._max && Number(this._to24(this._max).h);
            if (range === 'min' && this.min) {
                status = i < min;
            }
            if (range === 'max' && this.max) {
                status = i > max;
            }
            if (min > max) {
                status = false;
                status = min > i && i > max;
            }
        }
        this._disabledHours[i] = status;
        return status;
    };
    MdbTimePickerContentComponent.prototype._rangeMinute = function (index, range) {
        var status = false;
        if (!this._showHours) {
            if (range === 'min' && this.min) {
                var isSameHour = this._min.h === this._selectedTime.h;
                var value = index < Number(this._min.m);
                status = value && isSameHour;
            }
            else if (range === 'max' && this.max) {
                var isSameHour = this._max.h === this._selectedTime.h;
                var value = index > Number(this._max.m);
                status = value && isSameHour;
            }
        }
        if (this.twelveHour) {
            var min = this._min && Number(this._to24(this._min).h);
            var max = this._max && Number(this._to24(this._max).h);
            var i = Number(this._to24(this._selectedTime).h);
            if (range === 'min' && min) {
                status = i === min && index < Number(this._min.m);
            }
            else if (range === 'max' && max) {
                status = i === max && index > Number(this._max.m);
            }
            status = status || this._disabledHours[i];
        }
        return status;
    };
    MdbTimePickerContentComponent.prototype._setMinuteDigitalDisabled = function () {
        var h = this._to24(this._selectedTime).h;
        this._minuteDigitalDisabled = this._disabledHours[Number(h)];
    };
    MdbTimePickerContentComponent.ctorParameters = function () { return [
        { type: ChangeDetectorRef },
        { type: NgZone },
        { type: FocusMonitor },
        { type: ElementRef },
        { type: Renderer2 },
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
    ]; };
    __decorate([
        ViewChild('plate', { static: false }),
        __metadata("design:type", ElementRef)
    ], MdbTimePickerContentComponent.prototype, "plate", void 0);
    __decorate([
        ViewChild('hand', { static: false }),
        __metadata("design:type", ElementRef)
    ], MdbTimePickerContentComponent.prototype, "hand", void 0);
    __decorate([
        ViewChild('fg', { static: false }),
        __metadata("design:type", ElementRef)
    ], MdbTimePickerContentComponent.prototype, "fg", void 0);
    __decorate([
        ViewChild('bg', { static: false }),
        __metadata("design:type", ElementRef)
    ], MdbTimePickerContentComponent.prototype, "bg", void 0);
    __decorate([
        ViewChild('focus', { static: false }),
        __metadata("design:type", ElementRef)
    ], MdbTimePickerContentComponent.prototype, "focus", void 0);
    __decorate([
        ViewChild('digitalMinute', { static: false }),
        __metadata("design:type", ElementRef)
    ], MdbTimePickerContentComponent.prototype, "digitalMinute", void 0);
    MdbTimePickerContentComponent = __decorate([ __param(5, Inject(DOCUMENT)),
        __metadata("design:paramtypes", [ChangeDetectorRef,
            NgZone,
            FocusMonitor,
            ElementRef,
            Renderer2, Object])
    ], MdbTimePickerContentComponent);
MdbTimePickerContentComponent.ɵfac = function MdbTimePickerContentComponent_Factory(t) { return new (t || MdbTimePickerContentComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.NgZone), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FocusMonitor), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(DOCUMENT)); };
MdbTimePickerContentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MdbTimePickerContentComponent, selectors: [["mdb-timepicker-content"]], viewQuery: function MdbTimePickerContentComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
        ɵngcc0.ɵɵviewQuery(_c1, true);
        ɵngcc0.ɵɵviewQuery(_c2, true);
        ɵngcc0.ɵɵviewQuery(_c3, true);
        ɵngcc0.ɵɵviewQuery(_c4, true);
        ɵngcc0.ɵɵviewQuery(_c5, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.plate = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.hand = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.fg = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.bg = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.focus = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.digitalMinute = _t.first);
    } }, decls: 38, vars: 30, consts: [["cdkTrapFocus", "", 1, "mdb-timepicker-modal"], [1, "mdb-timepicker-header"], [1, "mdb-timepicker-time"], ["tabindex", "0", 1, "hour-digital", 3, "ngClass", "click", "keydown.arrowdown", "keydown.arrowup", "keydown.enter"], ["focus", ""], ["tabindex", "0", 1, "minute-digital", 3, "ngClass", "click", "keydown.arrowdown", "keydown.arrowup", "keydown.enter"], ["digitalMinute", ""], ["class", "mdb-timepicker-ampm", 4, "ngIf"], [1, "mdb-timepicker-body"], [1, "mdb-timepicker-plate"], ["plate", ""], [1, "mdb-timepicker-canvas"], ["width", "270", "height", "270", 1, "mdb-timepicker-svg"], ["svg", ""], ["transform", "translate(135,135)"], ["g", ""], ["x1", "0", "y1", "0", "x2", "0", "y2", "-90"], ["hand", ""], ["r", "5", "cx", "0", "cy", "-110", 1, "mdb-timepicker-canvas-fg"], ["fg", ""], ["r", "20", "cx", "0", "cy", "-110", 1, "mdb-timepicker-canvas-bg"], ["bg", ""], ["cx", "0", "cy", "0", "r", "2", 1, "mdb-timepicker-canvas-bearing"], ["bearing", ""], [1, "mdb-timepicker-dial", "mdb-timepicker-hours", 3, "ngClass", "ngStyle"], ["hoursPlate", ""], ["class", "mdb-timepicker-tick", "style", "font-size: 140%;", 3, "ngClass", "ngStyle", "id", 4, "ngFor", "ngForOf"], [1, "mdb-timepicker-dial", "mdb-timepicker-minutes", 3, "ngClass", "ngStyle"], ["minutesPlate", ""], ["class", "mdb-timepicker-tick", "style", "font-size: 120%;", 3, "ngClass", "ngStyle", 4, "ngFor", "ngForOf"], [1, "mdb-timepicker-footer"], ["class", "mdb-timepicker-btn mdb-timepicker-clear", "mdbWavesEffect", "", "type", "button", 3, "click", 4, "ngIf"], ["class", "mdb-timepicker-btn mdb-timepicker-close", "mdbWavesEffect", "", "type", "button", 3, "click", 4, "ngIf"], ["mdbWavesEffect", "", "type", "button", 1, "mdb-timepicker-btn", "mdb-timepicker-ok", 3, "ngClass", "click"], [1, "mdb-timepicker-ampm"], ["tabindex", "0", 3, "ngClass", "click", "keydown.enter"], [1, "mdb-timepicker-tick", 2, "font-size", "140%", 3, "ngClass", "ngStyle", "id"], [1, "mdb-timepicker-tick", 2, "font-size", "120%", 3, "ngClass", "ngStyle"], ["mdbWavesEffect", "", "type", "button", 1, "mdb-timepicker-btn", "mdb-timepicker-clear", 3, "click"], ["mdbWavesEffect", "", "type", "button", 1, "mdb-timepicker-btn", "mdb-timepicker-close", 3, "click"]], template: function MdbTimePickerContentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "span", 3, 4);
        ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_Template_span_click_3_listener() { return ctx._showHoursClock(); })("keydown.arrowdown", function MdbTimePickerContentComponent_Template_span_keydown_arrowdown_3_listener($event) { return ctx._arrowChangeHour($event.key); })("keydown.arrowup", function MdbTimePickerContentComponent_Template_span_keydown_arrowup_3_listener($event) { return ctx._arrowChangeHour($event.key); })("keydown.enter", function MdbTimePickerContentComponent_Template_span_keydown_enter_3_listener() { return ctx._showHoursClock(); });
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtext(6, ":");
        ɵngcc0.ɵɵelementStart(7, "span", 5, 6);
        ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_Template_span_click_7_listener() { return ctx._showMinutesClock(); })("keydown.arrowdown", function MdbTimePickerContentComponent_Template_span_keydown_arrowdown_7_listener($event) { return ctx._arrowChangeMinute($event.key); })("keydown.arrowup", function MdbTimePickerContentComponent_Template_span_keydown_arrowup_7_listener($event) { return ctx._arrowChangeMinute($event.key); })("keydown.enter", function MdbTimePickerContentComponent_Template_span_keydown_enter_7_listener() { return ctx._showMinutesClock(); });
        ɵngcc0.ɵɵtext(9);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(10, MdbTimePickerContentComponent_div_10_Template, 5, 6, "div", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 8);
        ɵngcc0.ɵɵelementStart(12, "div", 9, 10);
        ɵngcc0.ɵɵelementStart(14, "div", 11);
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelementStart(15, "svg", 12, 13);
        ɵngcc0.ɵɵelementStart(17, "g", 14, 15);
        ɵngcc0.ɵɵelement(19, "line", 16, 17);
        ɵngcc0.ɵɵelement(21, "circle", 18, 19);
        ɵngcc0.ɵɵelement(23, "circle", 20, 21);
        ɵngcc0.ɵɵelement(25, "circle", 22, 23);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵnamespaceHTML();
        ɵngcc0.ɵɵelementStart(27, "div", 24, 25);
        ɵngcc0.ɵɵtemplate(29, MdbTimePickerContentComponent_div_29_Template, 2, 9, "div", 26);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(30, "div", 27, 28);
        ɵngcc0.ɵɵtemplate(32, MdbTimePickerContentComponent_div_32_Template, 2, 8, "div", 29);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(33, "div", 30);
        ɵngcc0.ɵɵtemplate(34, MdbTimePickerContentComponent_button_34_Template, 2, 1, "button", 31);
        ɵngcc0.ɵɵtemplate(35, MdbTimePickerContentComponent_button_35_Template, 2, 1, "button", 32);
        ɵngcc0.ɵɵelementStart(36, "button", 33);
        ɵngcc0.ɵɵlistener("click", function MdbTimePickerContentComponent_Template_button_click_36_listener() { return ctx._okBtnClicked(); });
        ɵngcc0.ɵɵtext(37);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(15, _c6, ctx._showHours));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx._selectedTime.h, "");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(17, _c9, !ctx._showHours, ctx._minuteDigitalDisabled));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx._selectedTime.m);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.twelveHour);
        ɵngcc0.ɵɵadvance(17);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(20, _c10, !ctx._showHours))("ngStyle", ɵngcc0.ɵɵpureFunction1(22, _c11, !ctx._showHours ? "hidden" : "visible"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx._hoursTicks);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(24, _c10, ctx._showHours))("ngStyle", ɵngcc0.ɵɵpureFunction1(26, _c11, ctx._showHours ? "hidden" : "visible"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx._minutesTicks);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.clearButton);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.closeButton);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(28, _c7, ctx._okButtonDisabled));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.okButton, " ");
    } }, directives: [ɵngcc1.CdkTrapFocus, ɵngcc2.NgClass, ɵngcc2.NgIf, ɵngcc2.NgStyle, ɵngcc2.NgForOf, ɵngcc3.WavesDirective], styles: ["@-webkit-keyframes pulse{from,to{transform:scale3d(1,1,1)}50%{transform:scale3d(1.1,1.1,1.1)}}@keyframes pulse{from,to{transform:scale3d(1,1,1)}50%{transform:scale3d(1.1,1.1,1.1)}}.disabled{cursor:default;opacity:.5}.mdb-timepicker-modal{min-width:328px;box-shadow:rgba(0,0,0,.2) 0 11px 15px -7px,rgba(0,0,0,.14) 0 24px 38px 3px,rgba(0,0,0,.12) 0 9px 46px 8px;background:#fff;display:table-cell;vertical-align:middle}@media (min-height:28.875em){.mdb-timepicker-modal{display:block;border:1px solid #777;border-top-color:#898989;border-bottom-width:0;border-radius:5px 5px 0 0;box-shadow:0 .75rem 2.25rem 1rem rgba(0,0,0,.24)}}.mdb-timepicker-header{height:120px;box-sizing:border-box;background-color:#4285f4;padding:24px;display:flex;align-items:center;justify-content:center;color:rgba(255,255,255,.55)}.mdb-timepicker-time{font-size:4.375rem;color:rgba(255,255,255,.65)}.mdb-timepicker-ampm{margin-left:15px;font-size:18px;display:flex;flex-direction:column;justify-content:space-around;height:100%}span{cursor:pointer}span.active{color:#fff}.mdb-timepicker-tick{border-radius:50%;color:#666;line-height:2.5rem;text-align:center;width:2.5rem;height:2.5rem;position:absolute;cursor:pointer;transition:.3s;background-color:rgba(0,150,136,0)}.mdb-timepicker-tick:hover{background-color:rgba(0,150,136,.25)}.mdb-timepicker-footer{display:flex;justify-content:flex-end;width:100%;padding:12px}.mdb-timepicker-btn{padding:6px 8px;text-transform:uppercase;background:0 0;border:0;border-radius:4px;min-width:64px;transition:.3s}.mdb-timepicker-btn:not(.mdb-timepicker-btn.mdb-timepicker-clear){margin-left:8px}.mdb-timepicker-btn.mdb-timepicker-clear{margin-right:auto}.mdb-timepicker-btn:hover{background-color:rgba(0,150,136,.25)}.mdb-timepicker-plate{background-color:#eee;border-radius:50%;width:16.875rem;height:16.875rem;overflow:visible;position:relative;margin:1.25rem auto auto;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.mdb-timepicker-plate .mdb-timepicker-minutes{visibility:hidden}.mdb-timepicker-plate .mdb-timepicker-dial-out{opacity:0}.mdb-timepicker-plate .mdb-timepicker-hours.mdb-timepicker-dial-out{transform:scale(1.2,1.2)}.mdb-timepicker-plate .mdb-timepicker-minutes.mdb-timepicker-dial-out{transform:scale(.8,.8)}.mdb-timepicker-canvas,.mdb-timepicker-dial{width:16.875rem;height:16.875rem;position:absolute;left:-1px;top:-1px}.mdb-timepicker-dial{transition:transform 350ms,opacity 350ms}.mdb-timepicker-dial .mdb-timepicker-tick{border-radius:50%;color:#666;line-height:2.5rem;text-align:center;width:2.5rem;height:2.5rem;position:absolute;cursor:pointer;transition:background-color .3s;background-color:rgba(0,150,136,0)}.mdb-timepicker-dial .mdb-timepicker-tick.active,.mdb-timepicker-dial .mdb-timepicker-tick:hover{background-color:rgba(0,150,136,.25)}.mdb-timepicker-canvas{transition:opacity .3s}.mdb-timepicker-canvas line{stroke:rgba(0,150,136,.25);stroke-width:1}.mdb-timepicker-canvas-out{opacity:.25}.mdb-timepicker-canvas-bearing{stroke:none;fill:rgba(0,77,64,.75)}.mdb-timepicker-canvas-fg{stroke:none;fill:rgba(0,77,64,0)}.mdb-timepicker-canvas-fg.active{fill:rgba(0,77,64,.5)}.mdb-timepicker-canvas-bg{stroke:none;fill:rgba(0,150,136,.25)}.mdb-timepicker-canvas-bg-trans{fill:rgba(0,150,136,.25)}"], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbTimePickerContentComponent, [{
        type: Component,
        args: [{
                selector: 'mdb-timepicker-content',
                template: "<div class=\"mdb-timepicker-modal\" cdkTrapFocus>\n  <!-- HEADER -->\n  <div class=\"mdb-timepicker-header\">\n    <!-- TIME -->\n    <div class=\"mdb-timepicker-time\">\n      <span\n        (click)=\"_showHoursClock()\"\n        (keydown.arrowdown)=\"_arrowChangeHour($event.key)\"\n        (keydown.arrowup)=\"_arrowChangeHour($event.key)\"\n        (keydown.enter)=\"_showHoursClock()\"\n        [ngClass]=\"{ active: _showHours }\"\n        class=\"hour-digital\"\n        #focus\n        tabindex=\"0\"\n      >\n        {{ _selectedTime.h }}</span\n      >:<span\n        (click)=\"_showMinutesClock()\"\n        (keydown.arrowdown)=\"_arrowChangeMinute($event.key)\"\n        (keydown.arrowup)=\"_arrowChangeMinute($event.key)\"\n        (keydown.enter)=\"_showMinutesClock()\"\n        [ngClass]=\"{ 'active': !_showHours, 'disabled': _minuteDigitalDisabled }\"\n        class=\"minute-digital\"\n        #digitalMinute\n        tabindex=\"0\"\n        >{{ _selectedTime.m }}</span\n      >\n    </div>\n    <div class=\"mdb-timepicker-ampm\" *ngIf=\"twelveHour\">\n      <span\n        (click)=\"_setAmPm('AM')\"\n        (keydown.enter)=\"_setAmPm('AM')\"\n        [ngClass]=\"{ active: _selectedTime.ampm == 'AM' }\"\n        tabindex=\"0\"\n        >AM</span\n      >\n      <span\n        (click)=\"_setAmPm('PM')\"\n        (keydown.enter)=\"_setAmPm('PM')\"\n        [ngClass]=\"{ active: _selectedTime.ampm == 'PM' }\"\n        tabindex=\"0\"\n        >PM</span\n      >\n    </div>\n  </div>\n  <!-- /Header -->\n  <!-- Body -->\n  <div class=\"mdb-timepicker-body\">\n    <div class=\"mdb-timepicker-plate\" #plate>\n      <div class=\"mdb-timepicker-canvas\">\n        <svg class=\"mdb-timepicker-svg\" width=\"270\" height=\"270\" #svg>\n          <g transform=\"translate(135,135)\" #g>\n            <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"-90\" #hand></line>\n            <circle class=\"mdb-timepicker-canvas-fg\" r=\"5\" cx=\"0\" cy=\"-110\" #fg></circle>\n            <circle class=\"mdb-timepicker-canvas-bg\" r=\"20\" cx=\"0\" cy=\"-110\" #bg></circle>\n            <circle class=\"mdb-timepicker-canvas-bearing\" cx=\"0\" cy=\"0\" r=\"2\" #bearing></circle>\n          </g>\n        </svg>\n      </div>\n\n      <div\n        [ngClass]=\"{ 'mdb-timepicker-dial-out': !_showHours }\"\n        [ngStyle]=\"{ visibility: !_showHours ? 'hidden' : 'visible' }\"\n        #hoursPlate\n        class=\"mdb-timepicker-dial mdb-timepicker-hours\"\n      >\n        <div\n          [ngClass]=\"{ disabled: tick.disabled }\"\n          [ngStyle]=\"{ left: tick.left + 'px', top: tick.top + 'px' }\"\n          *ngFor=\"let tick of _hoursTicks\"\n          class=\"mdb-timepicker-tick\"\n          id=\"{{ tick.hour }}\"\n          style=\"font-size: 140%;\"\n        >\n          {{ tick.hour }}\n        </div>\n      </div>\n      <div\n        [ngClass]=\"{ 'mdb-timepicker-dial-out': _showHours }\"\n        [ngStyle]=\"{ visibility: _showHours ? 'hidden' : 'visible' }\"\n        #minutesPlate\n        class=\"mdb-timepicker-dial mdb-timepicker-minutes\"\n      >\n        <div\n          [ngClass]=\"{ disabled: tick.disabled }\"\n          [ngStyle]=\"{ left: tick.left + 'px', top: tick.top + 'px' }\"\n          *ngFor=\"let tick of _minutesTicks\"\n          class=\"mdb-timepicker-tick\"\n          style=\"font-size: 120%;\"\n        >\n          {{ tick.min }}\n        </div>\n      </div>\n    </div>\n  </div>\n  <!-- /Boody -->\n  <!-- Footer -->\n  <div class=\"mdb-timepicker-footer\">\n    <button\n      (click)=\"_clearBtnClicked()\"\n      *ngIf=\"clearButton\"\n      class=\"mdb-timepicker-btn mdb-timepicker-clear\"\n      mdbWavesEffect\n      type=\"button\"\n    >\n      {{ clearButton }}\n    </button>\n    <button\n      (click)=\"_closeBtnClicked()\"\n      *ngIf=\"closeButton\"\n      class=\"mdb-timepicker-btn mdb-timepicker-close\"\n      mdbWavesEffect\n      type=\"button\"\n    >\n      {{ closeButton }}\n    </button>\n    <button\n      (click)=\"_okBtnClicked()\"\n      [ngClass]=\"{ disabled: _okButtonDisabled }\"\n      class=\"mdb-timepicker-btn mdb-timepicker-ok\"\n      mdbWavesEffect\n      type=\"button\"\n    >\n      {{ okButton }}\n    </button>\n  </div>\n  <!-- /Footer -->\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: ["@-webkit-keyframes pulse{from,to{transform:scale3d(1,1,1)}50%{transform:scale3d(1.1,1.1,1.1)}}@keyframes pulse{from,to{transform:scale3d(1,1,1)}50%{transform:scale3d(1.1,1.1,1.1)}}.disabled{cursor:default;opacity:.5}.mdb-timepicker-modal{min-width:328px;box-shadow:rgba(0,0,0,.2) 0 11px 15px -7px,rgba(0,0,0,.14) 0 24px 38px 3px,rgba(0,0,0,.12) 0 9px 46px 8px;background:#fff;display:table-cell;vertical-align:middle}@media (min-height:28.875em){.mdb-timepicker-modal{display:block;border:1px solid #777;border-top-color:#898989;border-bottom-width:0;border-radius:5px 5px 0 0;box-shadow:0 .75rem 2.25rem 1rem rgba(0,0,0,.24)}}.mdb-timepicker-header{height:120px;box-sizing:border-box;background-color:#4285f4;padding:24px;display:flex;align-items:center;justify-content:center;color:rgba(255,255,255,.55)}.mdb-timepicker-time{font-size:4.375rem;color:rgba(255,255,255,.65)}.mdb-timepicker-ampm{margin-left:15px;font-size:18px;display:flex;flex-direction:column;justify-content:space-around;height:100%}span{cursor:pointer}span.active{color:#fff}.mdb-timepicker-tick{border-radius:50%;color:#666;line-height:2.5rem;text-align:center;width:2.5rem;height:2.5rem;position:absolute;cursor:pointer;transition:.3s;background-color:rgba(0,150,136,0)}.mdb-timepicker-tick:hover{background-color:rgba(0,150,136,.25)}.mdb-timepicker-footer{display:flex;justify-content:flex-end;width:100%;padding:12px}.mdb-timepicker-btn{padding:6px 8px;text-transform:uppercase;background:0 0;border:0;border-radius:4px;min-width:64px;transition:.3s}.mdb-timepicker-btn:not(.mdb-timepicker-btn.mdb-timepicker-clear){margin-left:8px}.mdb-timepicker-btn.mdb-timepicker-clear{margin-right:auto}.mdb-timepicker-btn:hover{background-color:rgba(0,150,136,.25)}.mdb-timepicker-plate{background-color:#eee;border-radius:50%;width:16.875rem;height:16.875rem;overflow:visible;position:relative;margin:1.25rem auto auto;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.mdb-timepicker-plate .mdb-timepicker-minutes{visibility:hidden}.mdb-timepicker-plate .mdb-timepicker-dial-out{opacity:0}.mdb-timepicker-plate .mdb-timepicker-hours.mdb-timepicker-dial-out{transform:scale(1.2,1.2)}.mdb-timepicker-plate .mdb-timepicker-minutes.mdb-timepicker-dial-out{transform:scale(.8,.8)}.mdb-timepicker-canvas,.mdb-timepicker-dial{width:16.875rem;height:16.875rem;position:absolute;left:-1px;top:-1px}.mdb-timepicker-dial{transition:transform 350ms,opacity 350ms}.mdb-timepicker-dial .mdb-timepicker-tick{border-radius:50%;color:#666;line-height:2.5rem;text-align:center;width:2.5rem;height:2.5rem;position:absolute;cursor:pointer;transition:background-color .3s;background-color:rgba(0,150,136,0)}.mdb-timepicker-dial .mdb-timepicker-tick.active,.mdb-timepicker-dial .mdb-timepicker-tick:hover{background-color:rgba(0,150,136,.25)}.mdb-timepicker-canvas{transition:opacity .3s}.mdb-timepicker-canvas line{stroke:rgba(0,150,136,.25);stroke-width:1}.mdb-timepicker-canvas-out{opacity:.25}.mdb-timepicker-canvas-bearing{stroke:none;fill:rgba(0,77,64,.75)}.mdb-timepicker-canvas-fg{stroke:none;fill:rgba(0,77,64,0)}.mdb-timepicker-canvas-fg.active{fill:rgba(0,77,64,.5)}.mdb-timepicker-canvas-bg{stroke:none;fill:rgba(0,150,136,.25)}.mdb-timepicker-canvas-bg-trans{fill:rgba(0,150,136,.25)}"]
            }]
    }], function () { return [{ type: ɵngcc0.ChangeDetectorRef }, { type: ɵngcc0.NgZone }, { type: ɵngcc1.FocusMonitor }, { type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }]; }, { plate: [{
            type: ViewChild,
            args: ['plate', { static: false }]
        }], hand: [{
            type: ViewChild,
            args: ['hand', { static: false }]
        }], fg: [{
            type: ViewChild,
            args: ['fg', { static: false }]
        }], bg: [{
            type: ViewChild,
            args: ['bg', { static: false }]
        }], focus: [{
            type: ViewChild,
            args: ['focus', { static: false }]
        }], digitalMinute: [{
            type: ViewChild,
            args: ['digitalMinute', { static: false }]
        }] }); })();
    return MdbTimePickerContentComponent;
}());
export { MdbTimePickerContentComponent };
