// ng-uikit-pro-standard/esm5/lib/pro/timepicker/timepicker.module.js
import { __decorate } from "tslib";
import { OverlayModule } from '@angular/cdk/overlay';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { A11yModule } from '@angular/cdk/a11y';
import { ButtonsModule } from '../../free/buttons/buttons.module';
import { WavesModule } from '../../free/waves/waves.module';
import { MdbTimepickerToggleComponent } from './timepicker-toggle.component';
import { MdbTimePickerDirective } from './timepicker.directive';
import { MdbTimePickerComponent } from './timepicker.component';
import { MdbTimePickerContentComponent } from './timepicker.content';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../free/buttons/buttons.module';
import * as ɵngcc2 from '../../free/waves/waves.module';
var MdbTimePickerModule = /** @class */ (function () {
    function MdbTimePickerModule() {
    }
MdbTimePickerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MdbTimePickerModule, bootstrap: function () { return [MdbTimePickerContentComponent]; } });
MdbTimePickerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MdbTimePickerModule_Factory(t) { return new (t || MdbTimePickerModule)(); }, imports: [[
            CommonModule,
            OverlayModule,
            A11yModule,
            ButtonsModule.forRoot(),
            WavesModule.forRoot(),
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MdbTimePickerModule, { declarations: function () { return [MdbTimePickerComponent,
        MdbTimepickerToggleComponent,
        MdbTimePickerDirective,
        MdbTimePickerContentComponent]; }, imports: function () { return [CommonModule,
        OverlayModule,
        A11yModule, ɵngcc1.ButtonsModule, ɵngcc2.WavesModule]; }, exports: function () { return [MdbTimePickerComponent, MdbTimepickerToggleComponent, MdbTimePickerDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbTimePickerModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    OverlayModule,
                    A11yModule,
                    ButtonsModule.forRoot(),
                    WavesModule.forRoot(),
                ],
                declarations: [
                    MdbTimePickerComponent,
                    MdbTimepickerToggleComponent,
                    MdbTimePickerDirective,
                    MdbTimePickerContentComponent,
                ],
                exports: [MdbTimePickerComponent, MdbTimepickerToggleComponent, MdbTimePickerDirective],
                bootstrap: [MdbTimePickerContentComponent],
                entryComponents: [MdbTimePickerContentComponent]
            }]
    }], function () { return []; }, null); })();
    return MdbTimePickerModule;
}());
export { MdbTimePickerModule };
