// ng-uikit-pro-standard/esm2015/lib/pro/sidenav/sidenav.component.js
import { Component, ElementRef, HostListener, Inject, Input, PLATFORM_ID, Renderer2, ViewChild, ViewEncapsulation, ChangeDetectionStrategy, ChangeDetectorRef, } from '@angular/core';
import { isPlatformBrowser, DOCUMENT } from '@angular/common';
import { window } from './../../free/utils/facade/browser';
export class SidenavComponent {
    constructor(platformId, el, renderer, _cdRef, document) {
        this.el = el;
        this.renderer = renderer;
        this._cdRef = _cdRef;
        this.document = document;
        this.slimSidenav = false;
        this.isBrowser = false;
        this._sidenavTransform = 'translateX(-100%)';
        this.fixed = true;
        this._side = 'left';
        this.isBrowser = isPlatformBrowser(platformId);
    }
    get side() {
        return this._side;
    }
    set side(position) {
        if (position === 'left') {
            this._sidenavTransform = 'translateX(-100%)';
            this.renderer.removeClass(this.sideNav.nativeElement, 'side-nav-right');
        }
        else {
            this._sidenavTransform = 'translateX(100%)';
            this.renderer.addClass(this.sideNav.nativeElement, 'side-nav-right');
        }
        this._side = position;
    }
    ngOnInit() {
        if (this.sidenavBreakpoint && this.sidenavBreakpoint >= window.innerWidth) {
            this.hide();
        }
    }
    ngAfterViewInit() {
        if (this._side === 'right') {
            this.renderer.addClass(this.sideNav.nativeElement, 'side-nav-right');
        }
        if (this.isBrowser) {
            const sidenav = this.el.nativeElement;
            const sidenavChildren = sidenav.children[0].children;
            const sidenavMask = this.el.nativeElement.querySelector('.sidenav-bg');
            let sidenavChildrenHeight = 0;
            if (sidenavMask) {
                for (let i = 0; i < sidenavChildren.length; i++) {
                    if (sidenavChildren[i].classList.contains('sidenav-bg')) {
                        continue;
                    }
                    else {
                        for (let j = 0; j < sidenavChildren[i].children.length; j++) {
                            sidenavChildrenHeight += sidenavChildren[i].children[j].scrollHeight;
                        }
                    }
                }
                this.renderer.setStyle(sidenavMask, 'min-height', sidenavChildrenHeight + 16 + 'px');
            }
            this.windwosWidth = window.innerWidth;
            if (this.sidenavBreakpoint) {
                if (this.fixed) {
                    this.renderer.addClass(this.document.body, 'fixed-sn');
                    if (this.windwosWidth < +this.sidenavBreakpoint + 1) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                    }
                    else {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.setShown(true);
                    }
                }
                else {
                    this.renderer.addClass(this.document.body, 'hidden-sn');
                    this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                    this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                    this.setShown(false);
                }
            }
            else {
                if (this.fixed) {
                    this.renderer.addClass(this.document.body, 'fixed-sn');
                    if (this.windwosWidth < 1441) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                    }
                    else {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.setShown(true);
                    }
                }
                else {
                    this.renderer.addClass(this.document.body, 'hidden-sn');
                    this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                    this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                    this.setShown(false);
                }
            }
        }
    }
    windowsResize() {
        if (this.isBrowser) {
            const currentWidth = this.windwosWidth;
            this.windwosWidth = window.innerWidth;
            if (this.sidenavBreakpoint) {
                if (this.fixed && this.windwosWidth !== currentWidth) {
                    if (this.windwosWidth < +this.sidenavBreakpoint + 1) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                    }
                    if (this.windwosWidth > +this.sidenavBreakpoint && this.shown) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.hideOverlay();
                        this.setShown(true);
                    }
                    else if (this.windwosWidth > +this.sidenavBreakpoint) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.hideOverlay();
                        this.setShown(true);
                    }
                }
                else {
                    if (this.windwosWidth > +this.sidenavBreakpoint && this.windwosWidth !== currentWidth) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.hideOverlay();
                        this.setShown(false);
                    }
                }
            }
            else {
                if (this.fixed && this.windwosWidth !== currentWidth) {
                    if (this.windwosWidth < 1441) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                    }
                    if (this.windwosWidth > 1440 && this.shown) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.hideOverlay();
                        this.setShown(true);
                    }
                    else if (this.windwosWidth > 1440) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.hideOverlay();
                        this.setShown(true);
                    }
                }
                else {
                    if (this.windwosWidth > 1440 && this.windwosWidth !== currentWidth) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.hideOverlay();
                        this.setShown(false);
                    }
                }
            }
        }
    }
    show() {
        if (this.isBrowser) {
            if (this.sidenavBreakpoint) {
                if (this.fixed) {
                    if (this.windwosWidth < +this.sidenavBreakpoint + 1) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.setShown(true);
                        this.showOverlay();
                    }
                    else {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.setShown(true);
                    }
                }
                else {
                    this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                    this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                    this.setShown(true);
                    this.showOverlay();
                }
            }
            else {
                if (this.fixed) {
                    if (this.windwosWidth < 1441) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.setShown(true);
                        this.showOverlay();
                    }
                    else {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                        this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                        this.setShown(true);
                    }
                }
                else {
                    this.renderer.setStyle(this.sideNav.nativeElement, 'transform', 'translateX(0%)');
                    this.renderer.setStyle(this.el.nativeElement, 'transform', 'translateX(0%)');
                    this.setShown(true);
                    this.showOverlay();
                }
            }
            this._cdRef.markForCheck();
        }
    }
    hide() {
        if (this.isBrowser) {
            if (this.sidenavBreakpoint) {
                if (this.fixed) {
                    if (this.windwosWidth < +this.sidenavBreakpoint + 1) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                        this.hideOverlay();
                    }
                    else {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                    }
                }
                else {
                    this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                    this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                    this.setShown(false);
                    this.hideOverlay();
                }
            }
            else {
                if (this.fixed) {
                    if (this.windwosWidth < 1441) {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                        this.hideOverlay();
                    }
                    else {
                        this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                        this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                        this.setShown(false);
                    }
                }
                else {
                    this.renderer.setStyle(this.sideNav.nativeElement, 'transform', this._sidenavTransform);
                    this.renderer.setStyle(this.el.nativeElement, 'transform', this._sidenavTransform);
                    this.setShown(false);
                    this.hideOverlay();
                }
            }
            this._cdRef.markForCheck();
        }
    }
    toggle() {
        if (this.shown) {
            this.hide();
        }
        else {
            this.show();
        }
    }
    toggleSlim() {
        const sidenavOverlay = this.el.nativeElement.querySelector('.sidenav-bg');
        const linksHeading = this.el.nativeElement.querySelectorAll('mdb-accordion-item-head');
        this.slimSidenav = !this.slimSidenav;
        linksHeading.forEach((el) => {
            if (this.slimSidenav) {
                this.renderer.addClass(el, 'overflow-hidden');
            }
            else {
                this.renderer.removeClass(el, 'overflow-hidden');
            }
        });
        this.renderer.addClass(this.sideNav.nativeElement, 'overflow-hidden');
        if (this.slimSidenav) {
            this.renderer.addClass(this.sideNav.nativeElement, 'slim');
            this.renderer.addClass(sidenavOverlay, 'slim');
        }
        else {
            this.renderer.removeClass(this.sideNav.nativeElement, 'slim');
            this.renderer.removeClass(sidenavOverlay, 'slim');
        }
        this._cdRef.markForCheck();
    }
    showOverlay() {
        this.renderer.setStyle(this.overlay.nativeElement, 'display', 'block');
        setTimeout(() => {
            this.renderer.setStyle(this.overlay.nativeElement, 'opacity', '1');
        }, 10);
    }
    hideOverlay() {
        this.renderer.setStyle(this.overlay.nativeElement, 'opacity', '0');
        setTimeout(() => {
            this.renderer.setStyle(this.overlay.nativeElement, 'display', 'none');
        }, 200);
    }
    setShown(value) {
        setTimeout(() => {
            this.shown = value;
        }, 510);
    }
    ngOnDestroy() {
        if (this.document.body.classList.contains('hidden-sn')) {
            this.renderer.removeClass(this.document.body, 'hidden-sn');
        }
        else if (this.document.body.classList.contains('fixed-sn')) {
            this.renderer.removeClass(this.document.body, 'fixed-sn');
        }
    }
}
SidenavComponent.decorators = [
    { type: Component, args: [{
                selector: 'mdb-sidenav, mdb-side-nav',
                template: "<ul #sidenav id=\"slide-out\" class=\"{{ class }} side-nav\">\n  <ng-content></ng-content>\n</ul>\n\n<div\n  (click)=\"hide()\"\n  (touchstart)=\"hide()\"\n  #overlay\n  id=\"sidenav-overlay\"\n  style=\"display: none; opacity: 0\"\n></div>\n",
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [".sn-bg-1{background-image:url(https://mdbootstrap.com/img/Photos/Others/sidenav1.jpg)}.sn-bg-2{background-image:url(https://mdbootstrap.com/img/Photos/Others/sidenav2.jpg)}.sn-bg-3{background-image:url(https://mdbootstrap.com/img/Photos/Others/sidenav3.jpg)}.sn-bg-4{background-image:url(https://mdbootstrap.com/img/Photos/Others/sidenav4.jpg)}.side-nav{-webkit-backface-visibility:hidden;backface-visibility:hidden;background-position:50%;background-repeat:no-repeat;background-size:cover;box-shadow:0 2px 5px 0 rgba(0,0,0,.16),0 2px 10px 0 rgba(0,0,0,.12);color:#fff;height:100%;left:0;list-style-type:none;margin:0;overflow:hidden;padding:0;position:fixed;top:0;transform:translateX(-100%);width:15rem;will-change:transform;z-index:1040}.side-nav.wide{transition-property:top,bottom,width;width:15rem}.side-nav.wide .logo-wrapper.sn-ad-avatar-wrapper{border-bottom:1px solid hsla(0,0%,100%,.65);height:5rem;padding:19px 10px}.side-nav.wide .logo-wrapper.sn-ad-avatar-wrapper a{color:#fff;line-height:2.6rem}.side-nav.wide .logo-wrapper.sn-ad-avatar-wrapper a span{margin-top:-1rem;padding-left:.7rem}.side-nav.wide .logo-wrapper.sn-ad-avatar-wrapper a img{max-width:2.5rem;padding:0}.side-nav.wide .collapsible a{padding-left:23px;transition:all .3s ease-in-out}.side-nav.wide .collapsible a .sv-normal{opacity:1;transition:all .1s linear}.side-nav.wide .collapsible a .sv-slim{display:none;opacity:0;transition:all .1s linear}.side-nav.wide .collapsible .sv-slim-icon{height:36px;margin-right:0;padding-left:0;text-align:left;width:30px}.side-nav.wide .collapsible-body a{padding-left:47px}.side-nav.wide .fa-angle-down.rotate-icon{display:block;opacity:1}.side-nav.wide.slim{right:3.75rem;transition-duration:.2s,.2s,.35s;transition-property:top,bottom,width;transition-timing-function:linear,linear,ease;width:3.75rem}.side-nav.wide.slim .collapsible a{padding-left:23px;transition:all .3s ease-in-out}.side-nav.wide.slim .collapsible a .sv-normal{opacity:0;transition:all .1s linear}.side-nav.wide.slim .collapsible a .sv-slim{display:block;opacity:1;transition:all .1s linear}.side-nav.wide.slim .collapsible .sv-slim-icon{height:36px;margin-right:0;padding-left:0;text-align:left;width:30px}.side-nav.wide.slim .fa-angle-down.rotate-icon{display:none;opacity:0}.side-nav>ul{max-height:100vh}.side-nav ul{list-style:none;list-style-type:none;padding-left:0}.side-nav ul li{padding:0}.side-nav.right-aligned{left:auto;right:0;transform:translateX(100%)}.side-nav.side-nav-light .logo-wrapper{border-bottom:1px solid hsla(0,0%,60%,.3)}.side-nav.side-nav-light .logo-wrapper.sn-ad-avatar-wrapper{color:#555}.side-nav.side-nav-light .about{border-bottom:1px solid hsla(0,0%,60%,.3)}.side-nav.side-nav-light .about p{color:#555}.side-nav.side-nav-light .social{border-bottom:1px solid hsla(0,0%,60%,.3)}.side-nav.side-nav-light .social .fab,.side-nav.side-nav-light .social .far,.side-nav.side-nav-light .social .fas{color:#555}.side-nav.side-nav-light .search-form input[type=text]{border-bottom-color:hsla(0,0%,60%,.3);color:#555!important}.side-nav.side-nav-light .search-form input[type=text]::-moz-placeholder{color:#555!important}.side-nav.side-nav-light .search-form input[type=text]::placeholder{color:#555!important}.side-nav.side-nav-light .collapsible a{color:#555}.side-nav.side-nav-light .collapsible-body a{background-color:rgba(0,0,0,.1)}.side-nav.side-nav-light .collapsible li .collapsible-header:hover{background-color:rgba(0,0,0,.05)}.side-nav.side-nav-light .collapsible li .collapsible-header.active{background-color:transparent;color:#4285f4}.side-nav.fixed{left:0;position:fixed;transform:translateX(0)}.side-nav.fixed.right-aligned{left:auto;right:0}@media only screen and (max-width:1440px){.side-nav.fixed{transform:translateX(-105%)}.side-nav.fixed.right-aligned{transform:translateX(105%)}}.side-nav .collapsible{margin:1rem 0 0;padding:0}.side-nav .collapsible li a:hover{background-color:rgba(0,0,0,.15)}.side-nav .collapsible>li{border-radius:2px}.side-nav .collapsible>li a.collapsible-header.active,.side-nav .collapsible>li a.collapsible-header:hover{background-color:hsla(0,0%,100%,.15)}.side-nav .collapsible ul{list-style-type:none;padding:0}.side-nav .collapsible a{color:#fff;font-size:.8rem;font-weight:300;height:36px;line-height:36px}.side-nav .collapsible a.active,.side-nav .collapsible a:hover{border-radius:2px}.side-nav .collapsible .fab,.side-nav .collapsible .far,.side-nav .collapsible .fas{font-size:.8rem;margin-right:13px}.side-nav .collapsible-body a{background-color:rgba(0,0,0,.15);height:36px;line-height:36px;padding-left:47px}.side-nav a{display:block;font-size:1rem;height:56px;line-height:56px;padding-left:20px}.side-nav .fa-angle-down.rotate-icon{margin-right:1.25rem;position:absolute;right:0;top:.8rem}.side-nav .sidenav-bg{background-attachment:fixed;bottom:0;left:0;right:0;top:0;width:15rem;z-index:-1}.side-nav .sidenav-bg:after{bottom:0;content:\"\";display:block;height:100%;left:0;margin-bottom:-99999px;padding-bottom:99999px;position:absolute;top:0;width:100%;z-index:-1}.side-nav .logo-wrapper{border-bottom:1px solid hsla(0,0%,60%,.3);height:8.75rem}.side-nav .logo-wrapper a{height:8.75rem;padding:0;width:15rem}.side-nav .logo-wrapper img{height:auto;padding:20% 50px}@media (max-height:992px){.side-nav .logo-wrapper,.side-nav .logo-wrapper a{height:80px}.side-nav .logo-wrapper img{padding-bottom:7%;padding-top:7%}}.side-nav .logo-wrapper.sn-avatar-wrapper{padding:10% 33%}.side-nav .logo-wrapper.sn-avatar-wrapper img{box-shadow:0 2px 5px 0 rgba(0,0,0,.16),0 2px 10px 0 rgba(0,0,0,.12);margin:0;max-width:90px;padding:0}@media only screen and (max-height:992px){.side-nav .logo-wrapper.sn-avatar-wrapper{padding-left:40%;padding-right:40%}.side-nav .logo-wrapper.sn-avatar-wrapper img{max-width:50px}}.side-nav .logo-wrapper.sn-ad-avatar-wrapper{height:auto;margin-bottom:0}.side-nav .logo-wrapper.sn-ad-avatar-wrapper img{float:left;max-width:3.75rem;padding:1.25rem .63rem}.side-nav .logo-wrapper.sn-ad-avatar-wrapper p{font-size:.94rem;margin:0;padding-bottom:1.25rem;padding-top:1.25rem}.side-nav .about{border-bottom:1px solid hsla(0,0%,100%,.65);padding:1rem}.side-nav .about p{margin-bottom:0;text-align:center}.side-nav .social{border-bottom:1px solid hsla(0,0%,100%,.65);padding-top:0;text-align:center}.side-nav .social li{display:inline-block;margin:0;padding-bottom:.6rem;padding-top:.6rem}.side-nav .social a{margin:0;padding:0}.side-nav .social .fab,.side-nav .social .far,.side-nav .social .fas{color:#dbe4e7;font-size:.9rem;padding-left:.6rem;padding-right:.6rem;transition:.3s}.side-nav .social .fab:hover,.side-nav .social .far:hover,.side-nav .social .fas:hover{color:#afc4cb;transition:.3s}.side-nav .search-form{padding:0}.side-nav .search-form input[type=text]{border-bottom:1px solid hsla(0,0%,100%,.65);font-weight:300;padding-left:1.88rem}.side-nav .search-form input[type=text]::-moz-placeholder{color:#fff}.side-nav .search-form input[type=text]::placeholder{color:#fff}.side-nav .search-form .form-control{color:#fff;margin-bottom:0}.drag-target{height:100%;position:fixed;top:0;width:10px;z-index:998}#sidenav-overlay{background-color:rgba(0,0,0,.5);height:120vh;left:0;position:fixed;right:0;top:0;will-change:opacity;z-index:997}.transform-fix-input{transform:translateX(0)!important}#sidenav-overlay,.side-nav{transition:.5s}.side-nav{background-color:#2c2f34;overflow-y:auto}.side-nav .sidenav-bg{position:fixed}.side-nav .sidenav-bg:after{margin-bottom:unset;padding-bottom:unset}.side-nav.side-nav-right{left:auto;right:0!important}.side-nav.side-nav-light{background-color:#e5e5e5}.side-nav.side-nav-light .collapsible .card .card-header a h5{color:#555}.side-nav.side-nav-light .collapsible .card .card-header a:hover{background-color:rgba(0,0,0,.05)}.side-nav.side-nav-light .collapsible .card.active .card-header a{background-color:transparent}.side-nav.side-nav-light .collapsible .card.active .card-header a h5,.side-nav.side-nav-light .collapsible .card.active .card-header a h5 i{color:#4285f4}.side-nav.side-nav-light .collapsible .card-body li a{background-color:rgba(0,0,0,.1);color:#555}.side-nav.side-nav-light .collapsible .card-body li a:hover{background-color:rgba(0,0,0,.15)}.side-nav.side-nav-light a{color:#555;font-weight:400}.side-nav.side-nav-light a .fa{color:#555}.side-nav.side-nav-light input[type=text]{border-bottom-color:hsla(0,0%,60%,.3);color:#555!important}.side-nav.side-nav-light input[type=text]::-webkit-input-placeholder{color:#555!important}.side-nav.side-nav-light input[type=text]::-moz-placeholder{color:#555!important}.side-nav.side-nav-light input[type=text]:-ms-input-placeholder{color:#555!important}.side-nav.side-nav-light input[type=text]:-moz-placeholder{color:#555!important}.side-nav::-webkit-scrollbar{height:4px;width:4px}.side-nav::-webkit-scrollbar-button:end:increment,.side-nav::-webkit-scrollbar-button:start:decrement{background-color:transparent;display:block;height:0}.side-nav::-webkit-scrollbar-track-piece{background-color:transparent;border-bottom-left-radius:4px;border-bottom-right-radius:4px;border-radius:0}.side-nav::-webkit-scrollbar-thumb:vertical{background-color:#999;border-radius:4px;height:50px}.side-nav input[type=text]:focus:not([readonly]){border-bottom:1px solid #4285f4;box-shadow:0 1px 0 0 #4285f4}.side-nav .collapsible .no-collase .rotate-icon{display:none}.side-nav .collapsible .card{background-color:transparent;border-bottom:0!important}.side-nav .collapsible .card .card-header{padding:0!important}.side-nav .collapsible .card .card-header a{border-radius:2px;height:40px;line-height:40px;padding:0;transition:all .3s}.side-nav .collapsible .card .card-header a h5{color:#fff;font-size:.8rem;font-weight:300;line-height:40px;padding-left:20px;position:relative}.side-nav .collapsible .card .card-header a .fa{font-size:.8rem;margin-right:13px;padding:0}.side-nav .collapsible .card .card-header a .rotate-icon{margin-right:20px;top:13px}.side-nav .collapsible .card.active>mdb-accordion-item-head{background:hsla(0,0%,100%,.15)}.side-nav .collapsible .card.active .card-header a h5{font-weight:300}.side-nav .collapsible .card-body{padding:0!important}.side-nav .collapsible .card-body li a{background-color:rgba(0,0,0,.15);color:#fff;display:block;font-size:.8rem;font-weight:300;height:32px;line-height:32px;padding-left:47px}.side-nav .collapsible .card-body li:first-of-type{border-top:0!important;margin-top:0}.side-nav.wide{transition-duration:.2s,.2s,.35s;transition-property:all;transition-timing-function:linear,linear,ease}.side-nav.wide .slim{right:3.75rem;transition-duration:.2s,.2s,.35s;transition-property:top,bottom,width;transition-timing-function:linear,linear,ease;width:3.75rem}.side-nav.wide .slim .collapsible a{padding-left:23px;transition:all .3s ease-in-out}.side-nav.wide .slim .collapsible a .sv-normal{opacity:0;transition:all .1s linear}.side-nav.wide .slim .collapsible a .sv-slim{display:block;opacity:1;transition:all .1s linear}.side-nav.wide .slim .collapsible .sv-slim-icon{height:36px;margin-right:0;padding-left:0;text-align:left;width:30px}.side-nav.wide .slim .fa-angle-down.rotate-icon{display:none;opacity:0}.slim .mdb-accordion-indicator{display:none}doublenavbar #sidenav-overlay,doublenavbar .side-nav{transition:.2s}doublenavbar links,doublenavbar navlinks{display:flex}header{height:0!important}@media (min-width:992px){.double-nav logo{align-items:center!important;align-self:center!important;display:flex;flex-direction:row;width:100%!important}.double-nav .navbar-brand img{margin-top:2px}}@media (max-width:1440px){.double-nav .button-collapse{display:block;font-size:1.4rem;left:10px;margin-left:10px;margin-right:10px;margin-top:-2px!important;top:25%}}.double-nav .button-collapse-non-fixed{font-size:1.5rem;left:10px}@media (min-width:1440px){.double-nav .button-collapse-non-fixed{display:none}}@media (max-width:599px){.double-nav .button-collapse-non-fixed{margin-top:-6px}}@media (max-width:1440px){.double-nav .button-collapse-non-fixed{display:block;font-size:1.4rem;left:10px;margin-left:10px;margin-right:10px;top:15px}}@media (min-width:600px) and (max-width:993px){.double-nav .button-collapse-non-fixed{margin-top:5px;top:10px}}@media (min-width:600px) and (max-width:993px){.double-nav .button-collapse-non-fixed.hidden-nav>i{margin-top:-5px!important;top:5px!important}}.double-nav .hidden-nav-button-collapse{display:block!important}@media (min-width:600px) and (max-width:993px){.double-nav .hidden-nav-button-collapse{margin-top:-3px;top:25%}}@media (min-width:994px){.double-nav .hidden-nav-button-collapse{margin-top:-3px}}.double-nav .hidden-nav{display:block;font-size:1.5rem;left:10px;margin-left:10px;top:13px}.double-nav .dropdown .dropdown-menu{display:none;left:0;margin-left:-68px;position:absolute;top:0;transform:translate3d(6px,49px,0);will-change:transform}.double-nav .mega-menu{left:240px!important;margin-left:0!important;width:calc(100% - 240px)!important}@media (max-width:1440px){.double-nav .mega-menu{left:0!important;width:100%!important}}@media (-ms-high-contrast:active),(-ms-high-contrast:none){.ie-hidden-double-nav{box-sizing:content-box;max-height:40.5px}.breadcrumbs{margin-left:25px!important;margin-top:2px}.breadcrumbs img{margin-left:20px}.ie-double-nav{position:absolute;right:0;top:30%}nav.double-nav{min-height:50px!important}.nav-link{padding-bottom:0!important;padding-top:0!important}.button-collapse,.button-collapse-non-fixed{position:absolute;top:30%!important}.hidden-nav-button-collapse{position:absolute}}@media (-ms-high-contrast:active) and (max-width:599px),(-ms-high-contrast:none) and (max-width:599px){.ie-double-nav{margin-top:-4px}.ie-hidden-double-nav .ie-double-nav{margin-top:0}.button-collapse,.button-collapse-non-fixed{margin-left:0!important;margin-right:0!important;top:5px!important}}@media (-ms-high-contrast:active) and (min-width:600px) and (max-width:992px),(-ms-high-contrast:none) and (min-width:600px) and (max-width:992px){.button-collapse-non-fixed{margin-top:-4px!important}.button-collapse{margin-top:-6px!important}.ie-double-nav{margin-top:0!important}.hidden-nav-button-collapse i{margin-top:10px!important}.breadcrumbs p{margin-top:-8px!important}}@media (-ms-high-contrast:active) and (min-width:993px) and (max-width:1441px),(-ms-high-contrast:none) and (min-width:993px) and (max-width:1441px){.button-collapse,.button-collapse-non-fixed{margin-top:-6px!important}.breadcrumbs>p{margin-top:-2px!important}.navbar-brand>img{margin-top:-2px}.breadcrumbs-non-fixed p{margin-top:2px!important}.hidden-nav-button-collapse{margin-left:10px!important;position:absolute!important}}@media (-ms-high-contrast:active) and (min-width:1441px),(-ms-high-contrast:none) and (min-width:1441px){.hidden-nav-button-collapse{margin-left:10px!important;position:absolute!important}.button-collapse-non-fixed{margin-top:-5px}.breadcrumbs>.navbar-brand img{margin-top:-2px}}@supports (-ms-ime-align:auto){.ie-hidden-double-nav{box-sizing:content-box;max-height:40.5px}.breadcrumbs{margin-left:25px!important;margin-top:2px}.breadcrumbs img{margin-left:20px}.ie-double-nav{position:absolute;right:0;top:30%}nav.double-nav{min-height:50px!important}.nav-link{padding-bottom:0!important;padding-top:0!important}.button-collapse,.button-collapse-non-fixed{position:absolute;top:30%!important}.hidden-nav-button-collapse{position:absolute}@media (max-width:599px){.ie-double-nav{margin-top:-4px}.ie-hidden-double-nav .ie-double-nav{margin-top:0}.button-collapse,.button-collapse-non-fixed{margin-left:0!important;margin-right:0!important;top:5px!important}}@media (min-width:600px) and (max-width:992px){.button-collapse-non-fixed{margin-top:-4px!important}.button-collapse{margin-top:-6px!important}.ie-double-nav{margin-top:0!important}.hidden-nav-button-collapse i{margin-top:10px!important}.breadcrumbs p{margin-top:-8px!important}}@media (min-width:993px) and (max-width:1441px){.button-collapse,.button-collapse-non-fixed{margin-top:-6px!important}.breadcrumbs>p{margin-top:-2px!important}.breadcrumbs>.navbar-brand>img{margin-top:-2px}.breadcrumbs-non-fixed p{margin-top:2px!important}.hidden-nav-button-collapse{margin-left:10px!important;position:absolute!important}}@media (min-width:1441px){.hidden-nav-button-collapse{margin-left:10px!important;position:absolute!important}.button-collapse-non-fixed{margin-top:-5px}.breadcrumbs>.navbar-brand img{margin-top:-2px}}}"]
            },] }
];
SidenavComponent.ctorParameters = () => [
    { type: String, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] },
    { type: ElementRef },
    { type: Renderer2 },
    { type: ChangeDetectorRef },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
SidenavComponent.propDecorators = {
    class: [{ type: Input }],
    fixed: [{ type: Input }],
    sidenavBreakpoint: [{ type: Input }],
    side: [{ type: Input }],
    sideNav: [{ type: ViewChild, args: ['sidenav', { static: true },] }],
    overlay: [{ type: ViewChild, args: ['overlay', { static: true },] }],
    windowsResize: [{ type: HostListener, args: ['window:resize',] }]
};
