// ng-uikit-pro-standard/esm5/lib/pro/select/select-filter.component.js
import { __decorate, __metadata } from "tslib";
import { Component, OnInit, ElementRef, ViewChild, forwardRef, HostListener, EventEmitter, Output, Input, } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["input"];
export var MDB_SELECT_FILTER_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(function () { return MdbSelectFilterComponent; }),
    multi: true,
};
var MdbSelectFilterComponent = /** @class */ (function () {
    function MdbSelectFilterComponent(_el) {
        this._el = _el;
        this.placeholder = '';
        this.autocomplete = true;
        this.inputChange = new EventEmitter();
        this._onChange = function () { };
        this._onTouched = function () { };
    }
    MdbSelectFilterComponent.prototype._handleInput = function (event) {
        var valueChanged = this.value !== event.target.value;
        if (valueChanged) {
            this._onChange(event.target.value);
            this.inputChange.emit(event.target.value);
            this.value = event.target.value;
        }
    };
    MdbSelectFilterComponent.prototype.ngOnInit = function () { };
    MdbSelectFilterComponent.prototype.focus = function () {
        this.input.nativeElement.focus();
    };
    /** Control value accessor methods */
    MdbSelectFilterComponent.prototype.setDisabledState = function (isDisabled) {
        this._el.nativeElement.disabled = isDisabled;
    };
    MdbSelectFilterComponent.prototype.writeValue = function (value) {
        var _this = this;
        Promise.resolve(null).then(function () {
            _this._el.nativeElement.value = value;
        });
    };
    MdbSelectFilterComponent.prototype.registerOnChange = function (fn) {
        this._onChange = fn;
    };
    MdbSelectFilterComponent.prototype.registerOnTouched = function (fn) {
        this._onTouched = fn;
    };
    MdbSelectFilterComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    __decorate([
        ViewChild('input'),
        __metadata("design:type", ElementRef)
    ], MdbSelectFilterComponent.prototype, "input", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], MdbSelectFilterComponent.prototype, "placeholder", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], MdbSelectFilterComponent.prototype, "autocomplete", void 0);
    __decorate([
        Output(),
        __metadata("design:type", EventEmitter)
    ], MdbSelectFilterComponent.prototype, "inputChange", void 0);
    __decorate([
        HostListener('input', ['$event']),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], MdbSelectFilterComponent.prototype, "_handleInput", null);
    MdbSelectFilterComponent = __decorate([ __metadata("design:paramtypes", [ElementRef])
    ], MdbSelectFilterComponent);
MdbSelectFilterComponent.ɵfac = function MdbSelectFilterComponent_Factory(t) { return new (t || MdbSelectFilterComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
MdbSelectFilterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MdbSelectFilterComponent, selectors: [["mdb-select-filter"]], viewQuery: function MdbSelectFilterComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.input = _t.first);
    } }, hostBindings: function MdbSelectFilterComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("input", function MdbSelectFilterComponent_input_HostBindingHandler($event) { return ctx._handleInput($event); });
    } }, inputs: { placeholder: "placeholder", autocomplete: "autocomplete" }, outputs: { inputChange: "inputChange" }, features: [ɵngcc0.ɵɵProvidersFeature([MDB_SELECT_FILTER_VALUE_ACCESSOR])], decls: 4, vars: 3, consts: [[1, "mdb-select-filter", "md-form", "px-2"], ["filter", ""], ["type", "text", 1, "mdb-select-filter-input", "search", "form-control", "w-100", "d-block", 3, "placeholder"], ["input", ""]], template: function MdbSelectFilterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0, 1);
        ɵngcc0.ɵɵelement(2, "input", 2, 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("placeholder", ctx.placeholder);
        ɵngcc0.ɵɵattribute("autocomplete", ctx.autocomplete)("role", "searchbox");
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbSelectFilterComponent, [{
        type: Component,
        args: [{
                selector: 'mdb-select-filter',
                template: "<div #filter class=\"mdb-select-filter md-form px-2\">\n  <input\n    #input\n    [placeholder]=\"placeholder\"\n    [attr.autocomplete]=\"autocomplete\"\n    [attr.role]=\"'searchbox'\"\n    type=\"text\"\n    class=\"mdb-select-filter-input search form-control w-100 d-block\"\n  />\n</div>\n",
                providers: [MDB_SELECT_FILTER_VALUE_ACCESSOR]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { placeholder: [{
            type: Input
        }], autocomplete: [{
            type: Input
        }], inputChange: [{
            type: Output
        }], _handleInput: [{
            type: HostListener,
            args: ['input', ['$event']]
        }], input: [{
            type: ViewChild,
            args: ['input']
        }] }); })();
    return MdbSelectFilterComponent;
}());
export { MdbSelectFilterComponent };
