// ng-uikit-pro-standard/esm2015/lib/pro/timepicker/timepicker-toggle.component.js
import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input, HostListener, } from '@angular/core';
import { MdbTimePickerComponent } from './timepicker.component';
export class MdbTimepickerToggleComponent {
    constructor() { }
    handleClick() {
        this.mdbTimePickerToggle.open();
    }
    ngOnInit() { }
}
MdbTimepickerToggleComponent.decorators = [
    { type: Component, args: [{
                template: "<button class=\"mdb-timepicker-toggle\">\n  <svg\n    aria-hidden=\"true\"\n    focusable=\"false\"\n    data-prefix=\"fas\"\n    data-icon=\"clock\"\n    class=\"svg-inline--fa fa-clock fa-w-16\"\n    role=\"img\"\n    xmlns=\"http://www.w3.org/2000/svg\"\n    viewBox=\"0 0 512 512\"\n  >\n    <path\n      fill=\"currentColor\"\n      d=\"M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z\"\n    ></path>\n  </svg>\n</button>\n",
                selector: 'mdb-timepicker-toggle',
                encapsulation: ViewEncapsulation.None,
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [".mdb-timepicker-toggle{background-color:transparent;border-color:transparent;position:absolute;right:0;top:50%;transform:translateY(-50%)}.mdb-timepicker-toggle svg{height:1em}"]
            },] }
];
MdbTimepickerToggleComponent.ctorParameters = () => [];
MdbTimepickerToggleComponent.propDecorators = {
    mdbTimePickerToggle: [{ type: Input }],
    handleClick: [{ type: HostListener, args: ['click',] }]
};
