// ng-uikit-pro-standard/esm5/lib/mdb.module.js
import { __decorate } from "tslib";
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { MDBBootstrapModule } from './free/mdb-free.module';
import { MDBBootstrapModulePro } from './pro/mdb-pro.module';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './free/mdb-free.module';
import * as ɵngcc2 from './pro/mdb-pro.module';
export { MDBBootstrapModule } from './free/mdb-free.module';
export { MDBBootstrapModulePro } from './pro/mdb-pro.module';
var MODULES = [MDBBootstrapModule, MDBBootstrapModulePro];
var MDBRootModules = /** @class */ (function () {
    function MDBRootModules() {
    }
MDBRootModules.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MDBRootModules });
MDBRootModules.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MDBRootModules_Factory(t) { return new (t || MDBRootModules)(); }, providers: [], imports: [[MDBBootstrapModule.forRoot(), MDBBootstrapModulePro.forRoot()], MDBBootstrapModule, MDBBootstrapModulePro] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MDBRootModules, { imports: [ɵngcc1.MDBRootModule, ɵngcc2.MDBRootModulePro], exports: [MDBBootstrapModule, MDBBootstrapModulePro] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MDBRootModules, [{
        type: NgModule,
        args: [{
                imports: [MDBBootstrapModule.forRoot(), MDBBootstrapModulePro.forRoot()],
                exports: MODULES,
                providers: [],
                schemas: [NO_ERRORS_SCHEMA]
            }]
    }], function () { return []; }, null); })();
    return MDBRootModules;
}());
export { MDBRootModules };
var MDBBootstrapModulesPro = /** @class */ (function () {
    function MDBBootstrapModulesPro() {
    }
    MDBBootstrapModulesPro.forRoot = function () {
        return { ngModule: MDBRootModules };
    };
MDBBootstrapModulesPro.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MDBBootstrapModulesPro });
MDBBootstrapModulesPro.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MDBBootstrapModulesPro_Factory(t) { return new (t || MDBBootstrapModulesPro)(); }, imports: [MDBBootstrapModule, MDBBootstrapModulePro] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MDBBootstrapModulesPro, { exports: [MDBBootstrapModule, MDBBootstrapModulePro] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MDBBootstrapModulesPro, [{
        type: NgModule,
        args: [{ exports: MODULES }]
    }], function () { return []; }, null); })();
    return MDBBootstrapModulesPro;
}());
export { MDBBootstrapModulesPro };
