// ng-uikit-pro-standard/esm2015/lib/pro/file-input/directives/mdb-file-select.directive.js
import { ElementRef, EventEmitter, Input, Output, PLATFORM_ID, Inject, ViewEncapsulation, Component, } from '@angular/core';
import { isPlatformServer } from '@angular/common';
import { MDBUploaderService } from '../classes/mdb-uploader.class';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
// tslint:disable-next-line:component-class-suffix
import * as ɵngcc0 from '@angular/core';

const _c0 = ["mdbFileSelect", ""];
const _c1 = ["*"];
export class MDBFileSelectDirective {
    constructor(platform_id, elementRef) {
        this.platform_id = platform_id;
        this.elementRef = elementRef;
        this._destroy$ = new Subject();
        this.isServer = isPlatformServer(this.platform_id);
        this.fileListener = () => {
            this.upload.handleFiles(this.el.files);
            this.el.value = '';
        };
        this.uploadOutput = new EventEmitter();
    }
    ngOnInit() {
        if (this.isServer) {
            return;
        }
        const concurrency = (this.options && this.options.concurrency) || Number.POSITIVE_INFINITY;
        const allowedContentTypes = (this.options && this.options.allowedContentTypes) || ['*'];
        const maxUploads = (this.options && this.options.maxUploads) || Number.POSITIVE_INFINITY;
        this.upload = new MDBUploaderService(concurrency, allowedContentTypes, maxUploads);
        this.el = this.elementRef.nativeElement;
        this.el.addEventListener('change', this.fileListener, false);
        this.upload.serviceEvents.pipe(takeUntil(this._destroy$)).subscribe((event) => {
            this.uploadOutput.emit(event);
        });
        if (this.uploadInput instanceof EventEmitter) {
            this.upload.initInputEvents(this.uploadInput);
        }
    }
    ngOnDestroy() {
        if (this.isServer) {
            return;
        }
        if (this.el) {
            this.el.removeEventListener('change', this.fileListener, false);
        }
        if (this.uploadInput) {
            this.uploadInput.unsubscribe();
        }
        this._destroy$.next();
        this._destroy$.complete();
    }
}
MDBFileSelectDirective.ɵfac = function MDBFileSelectDirective_Factory(t) { return new (t || MDBFileSelectDirective)(ɵngcc0.ɵɵdirectiveInject(PLATFORM_ID), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
MDBFileSelectDirective.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MDBFileSelectDirective, selectors: [["", "mdbFileSelect", ""]], inputs: { uploadInput: "uploadInput", options: "options" }, outputs: { uploadOutput: "uploadOutput" }, attrs: _c0, ngContentSelectors: _c1, decls: 1, vars: 0, template: function MDBFileSelectDirective_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [".file-field{position:relative}.file-field .file-path-wrapper{height:2.5rem;overflow:hidden;padding-left:10px}.file-field input.file-path{height:36px}.file-field .btn{float:left}.file-field input[type=file]::-webkit-file-upload-button{display:none}.file-field{display:flex;flex-direction:row}.file-field .file-path-wrapper{display:flex;flex-grow:1;overflow:visible}.file-field .file-path-wrapper:after{clear:both;content:\"\"}.file-field input.file-path{background-color:transparent;border:none;border-bottom:1px solid #ccc;border-radius:0;box-shadow:none;box-sizing:content-box;font-size:1rem;height:2.1rem;outline:none;transition:all .3s;width:100%}.file-field .btn{display:flex;float:none;position:relative}.file-field .btn:hover{cursor:pointer}.file-field .btn input[type=file]{height:100%}.file-field .btn input[type=file] :hover,.file-field span{cursor:pointer}.file-field input[type=file]{bottom:0;cursor:pointer;filter:alpha(opacity=0);font-size:1px;left:0;margin:0;opacity:0;padding:0;position:absolute;right:0;top:0;width:100%}.file-field input[type=file]::-ms-value{display:none}.file-field input[type=file]::-ms-browse{width:100%}.btn-file{padding-top:1px}"], encapsulation: 2 });
MDBFileSelectDirective.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] },
    { type: ElementRef }
];
MDBFileSelectDirective.propDecorators = {
    uploadInput: [{ type: Input }],
    options: [{ type: Input }],
    uploadOutput: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MDBFileSelectDirective, [{
        type: Component,
        args: [{
                // tslint:disable-next-line:component-selector
                selector: '[mdbFileSelect]',
                template: '<ng-content></ng-content>',
                encapsulation: ViewEncapsulation.None,
                styles: [".file-field{position:relative}.file-field .file-path-wrapper{height:2.5rem;overflow:hidden;padding-left:10px}.file-field input.file-path{height:36px}.file-field .btn{float:left}.file-field input[type=file]::-webkit-file-upload-button{display:none}.file-field{display:flex;flex-direction:row}.file-field .file-path-wrapper{display:flex;flex-grow:1;overflow:visible}.file-field .file-path-wrapper:after{clear:both;content:\"\"}.file-field input.file-path{background-color:transparent;border:none;border-bottom:1px solid #ccc;border-radius:0;box-shadow:none;box-sizing:content-box;font-size:1rem;height:2.1rem;outline:none;transition:all .3s;width:100%}.file-field .btn{display:flex;float:none;position:relative}.file-field .btn:hover{cursor:pointer}.file-field .btn input[type=file]{height:100%}.file-field .btn input[type=file] :hover,.file-field span{cursor:pointer}.file-field input[type=file]{bottom:0;cursor:pointer;filter:alpha(opacity=0);font-size:1px;left:0;margin:0;opacity:0;padding:0;position:absolute;right:0;top:0;width:100%}.file-field input[type=file]::-ms-value{display:none}.file-field input[type=file]::-ms-browse{width:100%}.btn-file{padding-top:1px}"]
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [PLATFORM_ID]
            }] }, { type: ɵngcc0.ElementRef }]; }, { uploadOutput: [{
            type: Output
        }], uploadInput: [{
            type: Input
        }], options: [{
            type: Input
        }] }); })();
