// ng-uikit-pro-standard/esm5/lib/free/utils/keyboard-navigation.js
/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */
export var MAC_ENTER = 3;
export var BACKSPACE = 8;
export var TAB = 9;
export var NUM_CENTER = 12;
export var ENTER = 13;
export var SHIFT = 16;
export var CONTROL = 17;
export var ALT = 18;
export var PAUSE = 19;
export var CAPS_LOCK = 20;
export var ESCAPE = 27;
export var SPACE = 32;
export var PAGE_UP = 33;
export var PAGE_DOWN = 34;
export var END = 35;
export var HOME = 36;
export var LEFT_ARROW = 37;
export var UP_ARROW = 38;
export var RIGHT_ARROW = 39;
export var DOWN_ARROW = 40;
export var PLUS_SIGN = 43;
export var PRINT_SCREEN = 44;
export var INSERT = 45;
export var DELETE = 46;
export var ZERO = 48;
export var ONE = 49;
export var TWO = 50;
export var THREE = 51;
export var FOUR = 52;
export var FIVE = 53;
export var SIX = 54;
export var SEVEN = 55;
export var EIGHT = 56;
export var NINE = 57;
export var FF_SEMICOLON = 59; // Firefox (Gecko) fires this for semicolon instead of 186
export var FF_EQUALS = 61; // Firefox (Gecko) fires this for equals instead of 187
export var QUESTION_MARK = 63;
export var AT_SIGN = 64;
export var A = 65;
export var B = 66;
export var C = 67;
export var D = 68;
export var E = 69;
export var F = 70;
export var G = 71;
export var H = 72;
export var I = 73;
export var J = 74;
export var K = 75;
export var L = 76;
export var M = 77;
export var N = 78;
export var O = 79;
export var P = 80;
export var Q = 81;
export var R = 82;
export var S = 83;
export var T = 84;
export var U = 85;
export var V = 86;
export var W = 87;
export var X = 88;
export var Y = 89;
export var Z = 90;
export var META = 91; // WIN_KEY_LEFT
export var MAC_WK_CMD_LEFT = 91;
export var MAC_WK_CMD_RIGHT = 93;
export var CONTEXT_MENU = 93;
export var NUMPAD_ZERO = 96;
export var NUMPAD_ONE = 97;
export var NUMPAD_TWO = 98;
export var NUMPAD_THREE = 99;
export var NUMPAD_FOUR = 100;
export var NUMPAD_FIVE = 101;
export var NUMPAD_SIX = 102;
export var NUMPAD_SEVEN = 103;
export var NUMPAD_EIGHT = 104;
export var NUMPAD_NINE = 105;
export var NUMPAD_MULTIPLY = 106;
export var NUMPAD_PLUS = 107;
export var NUMPAD_MINUS = 109;
export var NUMPAD_PERIOD = 110;
export var NUMPAD_DIVIDE = 111;
export var F1 = 112;
export var F2 = 113;
export var F3 = 114;
export var F4 = 115;
export var F5 = 116;
export var F6 = 117;
export var F7 = 118;
export var F8 = 119;
export var F9 = 120;
export var F10 = 121;
export var F11 = 122;
export var F12 = 123;
export var NUM_LOCK = 144;
export var SCROLL_LOCK = 145;
export var FIRST_MEDIA = 166;
export var FF_MINUS = 173;
export var MUTE = 173; // Firefox (Gecko) fires 181 for MUTE
export var VOLUME_DOWN = 174; // Firefox (Gecko) fires 182 for VOLUME_DOWN
export var VOLUME_UP = 175; // Firefox (Gecko) fires 183 for VOLUME_UP
export var FF_MUTE = 181;
export var FF_VOLUME_DOWN = 182;
export var LAST_MEDIA = 183;
export var FF_VOLUME_UP = 183;
export var SEMICOLON = 186; // Firefox (Gecko) fires 59 for SEMICOLON
export var EQUALS = 187; // Firefox (Gecko) fires 61 for EQUALS
export var COMMA = 188;
export var DASH = 189; // Firefox (Gecko) fires 173 for DASH/MINUS
export var SLASH = 191;
export var APOSTROPHE = 192;
export var TILDE = 192;
export var OPEN_SQUARE_BRACKET = 219;
export var BACKSLASH = 220;
export var CLOSE_SQUARE_BRACKET = 221;
export var SINGLE_QUOTE = 222;
export var MAC_META = 224;
