// ng-uikit-pro-standard/esm5/lib/pro/stepper/stepper.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbStepComponent } from './step.component';
import { MdbStepperComponent } from './stepper.component';
import * as ɵngcc0 from '@angular/core';
var StepperModule = /** @class */ (function () {
    function StepperModule() {
    }
StepperModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: StepperModule });
StepperModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function StepperModule_Factory(t) { return new (t || StepperModule)(); }, imports: [[
            CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(StepperModule, { declarations: function () { return [MdbStepperComponent,
        MdbStepComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MdbStepperComponent,
        MdbStepComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(StepperModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    MdbStepperComponent,
                    MdbStepComponent
                ],
                exports: [
                    MdbStepperComponent,
                    MdbStepComponent
                ],
                imports: [
                    CommonModule,
                ]
            }]
    }], function () { return []; }, null); })();
    return StepperModule;
}());
export { StepperModule };
