// ng-uikit-pro-standard/esm5/lib/pro/auto-completer/auto-completer.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { MdbAutoCompleterComponent } from './components/mdb-auto-completer.component';
import { MdbOptionComponent } from './components/mdb-option.component';
import { MdbAutoCompleterDirective } from './directives/mdb-auto-completer.directive';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';
import { MdbAutoCompleterOptionDirective } from './directives/mdb-auto-completer-option.directive';
import * as ɵngcc0 from '@angular/core';
var AutoCompleterModule = /** @class */ (function () {
    function AutoCompleterModule() {
    }
AutoCompleterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AutoCompleterModule });
AutoCompleterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AutoCompleterModule_Factory(t) { return new (t || AutoCompleterModule)(); }, imports: [[CommonModule, HttpClientModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AutoCompleterModule, { declarations: function () { return [MdbAutoCompleterComponent,
        MdbOptionComponent,
        MdbAutoCompleterDirective,
        MdbAutoCompleterOptionDirective]; }, imports: function () { return [CommonModule, HttpClientModule, FormsModule]; }, exports: function () { return [MdbAutoCompleterComponent,
        MdbOptionComponent,
        MdbAutoCompleterDirective,
        MdbAutoCompleterOptionDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AutoCompleterModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, HttpClientModule, FormsModule],
                declarations: [
                    MdbAutoCompleterComponent,
                    MdbOptionComponent,
                    MdbAutoCompleterDirective,
                    MdbAutoCompleterOptionDirective,
                ],
                exports: [
                    MdbAutoCompleterComponent,
                    MdbOptionComponent,
                    MdbAutoCompleterDirective,
                    MdbAutoCompleterOptionDirective,
                ]
            }]
    }], function () { return []; }, null); })();
    return AutoCompleterModule;
}());
export { AutoCompleterModule };
