// ng-uikit-pro-standard/esm2015/lib/pro/select/select.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MdbSelectComponent } from './select.component';
import { MdbOptionModule } from './../option/option.module';
import { OverlayModule } from '@angular/cdk/overlay';
import { MdbSelectFilterComponent } from './select-filter.component';
import * as ɵngcc0 from '@angular/core';
export class MdbSelectModule {
}
MdbSelectModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MdbSelectModule });
MdbSelectModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MdbSelectModule_Factory(t) { return new (t || MdbSelectModule)(); }, imports: [[CommonModule, MdbOptionModule, OverlayModule], MdbOptionModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MdbSelectModule, { declarations: function () { return [MdbSelectComponent, MdbSelectFilterComponent]; }, imports: function () { return [CommonModule, MdbOptionModule, OverlayModule]; }, exports: function () { return [MdbSelectComponent, MdbSelectFilterComponent, MdbOptionModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbSelectModule, [{
        type: NgModule,
        args: [{
                declarations: [MdbSelectComponent, MdbSelectFilterComponent],
                imports: [CommonModule, MdbOptionModule, OverlayModule],
                exports: [MdbSelectComponent, MdbSelectFilterComponent, MdbOptionModule]
            }]
    }], null, null); })();
