// ng-uikit-pro-standard/esm2015/lib/free/popover/popover-container.component.js
import { ChangeDetectionStrategy, Component, HostBinding, Input, ViewEncapsulation, } from '@angular/core';
import { PopoverConfig } from './popover.config';
import { isBs3 } from '../utils/ng2-bootstrap-config';
export class PopoverContainerComponent {
    constructor(config) {
        this.show = '!isBs3';
        this.role = 'tooltip';
        Object.assign(this, config);
    }
    get isBs3() {
        return isBs3();
    }
    ngOnInit() {
        this.class =
            'popover-fadeIn popover in popover-' +
                this.placement +
                ' ' +
                this.placement +
                ' bs-popover-' +
                this.placement +
                ' ' +
                this.containerClass;
    }
}
PopoverContainerComponent.decorators = [
    { type: Component, args: [{
                selector: 'mdb-popover-container',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: `
    <h3 class="popover-header" [ngClass]="headerClass" *ngIf="title">{{ title }}</h3>
    <div class="popover-body" [ngClass]="bodyClass">
      <ng-content></ng-content>
    </div>
  `,
                encapsulation: ViewEncapsulation.None,
                styles: [".popover.bs-tether-element-attached-bottom,.popover.popover-top{margin-top:-10px}.popover.bs-tether-element-attached-bottom:after,.popover.bs-tether-element-attached-bottom:before,.popover.popover-top:after,.popover.popover-top:before{border-bottom-width:0;left:50%}.popover.bs-tether-element-attached-bottom:before,.popover.popover-top:before{border-top-color:rgba(0,0,0,.25);bottom:-11px;margin-left:-11px}.popover.bs-tether-element-attached-bottom:after,.popover.popover-top:after{border-top-color:#fff;bottom:-10px;margin-left:-10px}.popover.bs-tether-element-attached-left,.popover.popover-right{margin-left:10px}.popover.bs-tether-element-attached-left:after,.popover.bs-tether-element-attached-left:before,.popover.popover-right:after,.popover.popover-right:before{border-left-width:0;top:50%}.popover.bs-tether-element-attached-left:before,.popover.popover-right:before{border-right-color:rgba(0,0,0,.25);left:-11px;margin-top:-11px}.popover.bs-tether-element-attached-left:after,.popover.popover-right:after{border-right-color:#fff;left:-10px;margin-top:-10px}.popover.bs-tether-element-attached-top,.popover.popover-bottom{margin-top:10px}.popover.bs-tether-element-attached-top:after,.popover.bs-tether-element-attached-top:before,.popover.popover-bottom:after,.popover.popover-bottom:before{border-top-width:0;left:50%}.popover.bs-tether-element-attached-top:before,.popover.popover-bottom:before{border-bottom-color:rgba(0,0,0,.25);margin-left:-11px;top:-11px}.popover.bs-tether-element-attached-top:after,.popover.popover-bottom:after{border-bottom-color:#f7f7f7;margin-left:-10px;top:-10px}.popover.bs-tether-element-attached-top .popover-title:before,.popover.popover-bottom .popover-title:before{border-bottom:1px solid #f7f7f7;content:\"\";display:block;left:50%;margin-left:-10px;position:absolute;top:0;width:20px}.popover.bs-tether-element-attached-right,.popover.popover-left{margin-left:-10px}.popover.bs-tether-element-attached-right:after,.popover.bs-tether-element-attached-right:before,.popover.popover-left:after,.popover.popover-left:before{border-right-width:0;top:50%}.popover.bs-tether-element-attached-right:before,.popover.popover-left:before{border-left-color:rgba(0,0,0,.25);margin-top:-11px;right:-11px}.popover.bs-tether-element-attached-right:after,.popover.popover-left:after{border-left-color:#fff;margin-top:-10px;right:-10px}.popover:after,.popover:before{border-color:transparent;border-style:solid;display:block;height:0;position:absolute;width:0}.popover:before{border-width:11px;content:\"\"}.popover:after{border-width:10px;content:\"\"}@-webkit-keyframes fadeInPopover{0%{opacity:0}to{opacity:1}}@keyframes fadeInPopover{0%{opacity:0}to{opacity:1}}.popover-fadeIn{-webkit-animation-delay:.2s;-webkit-animation-duration:.2s;-webkit-animation-fill-mode:both;-webkit-animation-name:fadeInPopover;animation-delay:.2s;animation-duration:.2s;animation-fill-mode:both;animation-name:fadeInPopover}"]
            },] }
];
PopoverContainerComponent.ctorParameters = () => [
    { type: PopoverConfig }
];
PopoverContainerComponent.propDecorators = {
    placement: [{ type: Input }],
    title: [{ type: Input }],
    show: [{ type: HostBinding, args: ['class.show',] }],
    role: [{ type: HostBinding, args: ['attr.role',] }],
    class: [{ type: HostBinding, args: ['class',] }]
};
