// ng-uikit-pro-standard/esm5/lib/free/cards/mdb-card-image.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var MdbCardImageComponent = /** @class */ (function () {
    function MdbCardImageComponent() {
    }
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], MdbCardImageComponent.prototype, "src", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], MdbCardImageComponent.prototype, "alt", void 0);
MdbCardImageComponent.ɵfac = function MdbCardImageComponent_Factory(t) { return new (t || MdbCardImageComponent)(); };
MdbCardImageComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MdbCardImageComponent, selectors: [["mdb-card-img"]], inputs: { src: "src", alt: "alt" }, decls: 1, vars: 2, consts: [[1, "img-fluid", 3, "src", "alt"]], template: function MdbCardImageComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "img", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("src", ctx.src, ɵngcc0.ɵɵsanitizeUrl)("alt", ctx.alt);
    } }, encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MdbCardImageComponent, [{
        type: Component,
        args: [{
                selector: 'mdb-card-img',
                template: "<img class=\"img-fluid\" [src]=\"src\" [alt]=\"alt\">",
                changeDetection: ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, { src: [{
            type: Input
        }], alt: [{
            type: Input
        }] }); })();
    return MdbCardImageComponent;
}());
export { MdbCardImageComponent };
