// ng-uikit-pro-standard/esm5/lib/pro/progressbars/index.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
export { BarComponent } from './bar.component';
export { ProgressDirective } from './progress.directive';
export { ProgressbarComponent } from './progressbar.component';
export { ProgressbarModule } from './progressbar.module';
export { ProgressbarConfigComponent } from './progressbar.config.component';
export { ProgressSpinnerComponent } from './progress-spinner.component';
import { ProgressbarModule } from './progressbar.module';
import { MdProgressSpinnerModule } from './progress-spinner-module/index';
import { MdProgressBarModule } from './progress-bars-module/index';
var MATERIAL_MODULES = [MdProgressBarModule, MdProgressSpinnerModule, ProgressbarModule];
var PreloadersModule = /** @class */ (function () {
    function PreloadersModule() {
    }
    PreloadersModule = __decorate([
        NgModule({
            imports: [
                MdProgressBarModule.forRoot(),
                MdProgressSpinnerModule.forRoot(),
                ProgressbarModule.forRoot(),
            ],
            exports: MATERIAL_MODULES,
        })
    ], PreloadersModule);
    return PreloadersModule;
}());
export { PreloadersModule };
var ProgressBars = /** @class */ (function () {
    function ProgressBars() {
    }
    ProgressBars.forRoot = function () {
        return { ngModule: PreloadersModule };
    };
    ProgressBars = __decorate([
        NgModule({
            imports: MATERIAL_MODULES,
            exports: MATERIAL_MODULES,
        })
    ], ProgressBars);
    return ProgressBars;
}());
export { ProgressBars };
