// ng-uikit-pro-standard/esm5/lib/pro/material-select/select.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { SelectComponent } from './select.component';
import { SelectDropdownComponent } from './select-dropdown.component';
import * as ɵngcc0 from '@angular/core';
export { SELECT_VALUE_ACCESSOR, SelectComponent } from './select.component';
var SelectModule = /** @class */ (function () {
    function SelectModule() {
    }
SelectModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SelectModule });
SelectModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SelectModule_Factory(t) { return new (t || SelectModule)(); }, imports: [[
            CommonModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SelectModule, { declarations: function () { return [SelectComponent,
        SelectDropdownComponent]; }, imports: function () { return [CommonModule,
        FormsModule]; }, exports: function () { return [SelectComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SelectModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    SelectComponent,
                    SelectDropdownComponent
                ],
                exports: [
                    SelectComponent
                ],
                imports: [
                    CommonModule,
                    FormsModule
                ]
            }]
    }], function () { return []; }, null); })();
    return SelectModule;
}());
export { SelectModule };
