// ng-uikit-pro-standard/esm5/lib/pro/material-select/select-dropdown.component.js
import { __decorate, __metadata } from "tslib";
import { AfterViewInit, Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewChild, ViewEncapsulation, ElementRef, HostListener, Renderer2, ChangeDetectorRef, ChangeDetectionStrategy, SimpleChanges, OnDestroy, } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { OptionList } from './option-list';
import { Subject } from 'rxjs';
import { debounceTime, takeUntil, tap, map } from 'rxjs/operators';
import { A, NINE, Z, ZERO } from '../../free/utils/keyboard-navigation';
var SelectDropdownComponent = /** @class */ (function () {
    function SelectDropdownComponent(_elementRef, _renderer, cdRef) {
        this._elementRef = _elementRef;
        this._renderer = _renderer;
        this.cdRef = cdRef;
        this.customClass = '';
        this.visibleOptions = 4;
        this.selectAllLabel = 'Select all';
        this.outline = false;
        this.close = new EventEmitter();
        this.optionClicked = new EventEmitter();
        this.singleFilterClick = new EventEmitter();
        this.singleFilterInput = new EventEmitter();
        this.singleFilterKeydown = new EventEmitter();
        this.animationDone = new EventEmitter();
        this.animationStart = new EventEmitter();
        this.selectAll = new EventEmitter();
        this.disabledColor = '#fff';
        this.disabledTextColor = '9e9e9e';
        // Used in sliding-down animation
        this.state = 'invisible';
        this.startHeight = 0;
        this.endHeight = 45;
        this.hasOptionsItems = true;
        this._destroy = new Subject();
        this._pressedKeysStream = new Subject();
        this._pressedKeys = [];
        this.selectAllSelected = false;
        this.searchIndex = 0;
        this.previousKey = '';
    }
    SelectDropdownComponent.prototype.onWindowKeydown = function (event) {
        if ((event.keyCode >= A && event.keyCode <= Z) ||
            (event.keyCode >= ZERO && event.keyCode <= NINE)) {
            this._pressedKeysStream.next(String.fromCharCode(event.keyCode));
        }
    };
    SelectDropdownComponent.prototype.highlightOptionByTyping = function () {
        var _this = this;
        this._pressedKeysStream
            .pipe(tap(function (key) { return _this._pressedKeys.push(key); }), map(function () { return _this._pressedKeys.join('').toLocaleLowerCase(); }), debounceTime(200), takeUntil(this._destroy))
            .subscribe(function (searchKey) {
            var items = Array.from(_this.optionList['_options'])
                .filter(function (elem) { return !elem.group; })
                .filter(function (elem) { return !elem.disabled; })
                .map(function (el) { return el.wrappedOption.label || el.wrappedOption.value; });
            _this.navigateThroughArray(searchKey, items);
            _this.previousKey = searchKey;
        });
    };
    SelectDropdownComponent.prototype.navigateThroughArray = function (key, itemSource) {
        var _this = this;
        var items = itemSource.filter(function (el) {
            return el
                .toString()
                .toLowerCase()
                .startsWith(key.toString().toLowerCase());
        });
        if (this.searchIndex > items.length - 1 || key !== this.previousKey) {
            this.searchIndex = 0;
        }
        this.highlightedItem = this.optionList.filtered.find(function (el) { return el.wrappedOption.label === items[_this.searchIndex]; });
        this.searchIndex++;
        if (this.highlightedItem) {
            this.optionList.highlightOption(this.highlightedItem);
            this.cdRef.markForCheck();
        }
        this.moveHighlightedIntoView();
        this._pressedKeys = [];
    };
    /** Event handlers. **/
    SelectDropdownComponent.prototype.onkeyup = function () {
        this.hasOptionsItems = this.optionList.filtered.length > 0;
        this.updateSelectAllState();
    };
    SelectDropdownComponent.prototype.onkeydown = function () {
        this.setOptionHeight();
    };
    SelectDropdownComponent.prototype.ngOnInit = function () {
        this.updateSelectAllState();
        this.optionsReset();
        this.setDropdownHeight();
        this.setVisibleOptionsNumber();
        this.highlightOptionByTyping();
    };
    SelectDropdownComponent.prototype.setDropdownHeight = function () {
        var _this = this;
        this.optionList.options.filter(function (el) { return function () {
            if (el.icon) {
                _this._renderer.setStyle(_this.optionsList.nativeElement, 'height', _this.dropdownHeight + 8 + 'px');
            }
            else {
                _this._renderer.setStyle(_this.optionsList.nativeElement, 'height', _this.dropdownHeight + 'px');
            }
        }; });
    };
    SelectDropdownComponent.prototype.setVisibleOptionsNumber = function () {
        this._renderer.setStyle(this.optionsList.nativeElement, 'max-height', this.dropdownMaxHeight + 'px');
    };
    SelectDropdownComponent.prototype.setOptionHeight = function () {
        var _this = this;
        var optionsItems = Array.from(this.optionsList.nativeElement.firstElementChild.children);
        optionsItems.forEach(function (el) {
            var isCustomElement = el.classList.contains('custom-select-content');
            if (el.firstElementChild) {
                if (_this.optionHeight && el.firstElementChild.tagName !== 'IMG' && !isCustomElement) {
                    _this._renderer.setStyle(el.firstElementChild, 'height', _this.optionHeight + "px");
                }
                if (el.firstElementChild.tagName !== 'IMG' && !isCustomElement) {
                    _this._renderer.setStyle(el.firstElementChild, 'line-height', _this.optionHeight + "px");
                }
            }
        });
    };
    SelectDropdownComponent.prototype.ngOnChanges = function (changes) {
        if (changes.hasOwnProperty('optionList')) {
            this.optionsReset();
        }
        if (changes.hasOwnProperty('dropdownHeight')) {
            this.setDropdownHeight();
        }
        var container = this._elementRef.nativeElement.classList;
        setTimeout(function () {
            container.add('fadeInSelect');
        }, 200);
    };
    SelectDropdownComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        // Sliding-down animation
        this.endHeight = this.dropdownContent.nativeElement.clientHeight;
        this.state = this.state === 'invisible' ? 'visible' : 'invisible';
        this.cdRef.detectChanges();
        if (this.multiple) {
            var disabledElements = this._elementRef.nativeElement.querySelectorAll('.disabled.optgroup');
            for (var i = 0; i < disabledElements.length; i++) {
                this._renderer.setStyle(disabledElements[i].firstElementChild.lastElementChild, 'display', 'none');
            }
        }
        this.setOptionHeight();
        this.moveHighlightedIntoView();
        if (this.filterEnabled) {
            setTimeout(function () {
                _this.filterInput.nativeElement.focus();
            }, 0);
        }
    };
    // Filter input (single select).
    SelectDropdownComponent.prototype.onSingleFilterClick = function () {
        this.singleFilterClick.emit(null);
    };
    SelectDropdownComponent.prototype.onSingleFilterInput = function (event) {
        this.singleFilterInput.emit(event.target.value);
    };
    SelectDropdownComponent.prototype.onSingleFilterKeydown = function (event) {
        this.singleFilterKeydown.emit(event);
    };
    // Options list.
    SelectDropdownComponent.prototype.onOptionsWheel = function (event) {
        this.handleOptionsWheel(event);
    };
    SelectDropdownComponent.prototype.onOptionClick = function (option) {
        this.optionClicked.emit(option);
        this.updateSelectAllState();
    };
    /** Initialization. **/
    SelectDropdownComponent.prototype.optionsReset = function () {
        this.optionList.filter('');
        this.optionList.highlight();
    };
    /** View. **/
    SelectDropdownComponent.prototype.getOptionStyle = function (option) {
        if (option.highlighted || option.hovered) {
            var optionStyle = {};
            optionStyle['height.px'] = this.optionHeight;
            if (typeof this.highlightColor !== 'undefined') {
                optionStyle['background-color'] = this.highlightColor;
            }
            if (typeof this.highlightTextColor !== 'undefined') {
                optionStyle['color'] = this.highlightTextColor;
            }
            return optionStyle;
        }
        else {
            return {};
        }
    };
    SelectDropdownComponent.prototype.onSelectAllClick = function () {
        this.selectAllSelected = !this.selectAllSelected;
        this.selectAll.emit(this.selectAllSelected);
    };
    SelectDropdownComponent.prototype.updateSelectAllState = function () {
        var areAllSelected = this.optionList.filtered
            .filter(function (option) { return !option.disabled; })
            .every(function (option) {
            return option.selected ? true : false;
        });
        areAllSelected ? (this.selectAllSelected = true) : (this.selectAllSelected = false);
        this.cdRef.detectChanges();
    };
    SelectDropdownComponent.prototype.clearFilterInput = function () {
        if (this.filterEnabled) {
            this.filterInput.nativeElement.value = '';
        }
    };
    SelectDropdownComponent.prototype.onAnimationDone = function () {
        this.animationDone.emit();
    };
    SelectDropdownComponent.prototype.onAnimationStart = function () {
        this.animationStart.emit();
    };
    SelectDropdownComponent.prototype.moveHighlightedIntoView = function () {
        var listHeight;
        var list = this.optionsList.nativeElement;
        listHeight =
            this.multiple && this.enableSelectAll
                ? list.offsetHeight - this.optionHeight
                : list.offsetHeight;
        var itemIndex = this.optionList.getHighlightedIndex();
        if (itemIndex > -1) {
            var item = list.children[0].children[itemIndex];
            var itemHeight = item.offsetHeight;
            var itemTop = itemIndex * itemHeight;
            var itemBottom = itemTop + itemHeight;
            var viewTop = list.scrollTop;
            var viewBottom = viewTop + listHeight;
            if (itemBottom > viewBottom) {
                list.scrollTop = itemBottom - listHeight;
            }
            else if (itemTop < viewTop) {
                list.scrollTop = itemTop;
            }
        }
    };
    SelectDropdownComponent.prototype.handleOptionsWheel = function (e) {
        var div = this.optionsList.nativeElement;
        var atTop = div.scrollTop === 0;
        var atBottom = div.offsetHeight + div.scrollTop === div.scrollHeight;
        if (atTop && e.deltaY < 0) {
            e.preventDefault();
        }
        else if (atBottom && e.deltaY > 0) {
            e.preventDefault();
        }
    };
    SelectDropdownComponent.prototype.ngOnDestroy = function () {
        this._destroy.next();
        this._destroy.complete();
    };
    SelectDropdownComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 },
        { type: ChangeDetectorRef }
    ]; };
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SelectDropdownComponent.prototype, "filterEnabled", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SelectDropdownComponent.prototype, "filterAutocomplete", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SelectDropdownComponent.prototype, "highlightColor", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SelectDropdownComponent.prototype, "highlightTextColor", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], SelectDropdownComponent.prototype, "left", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SelectDropdownComponent.prototype, "multiple", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SelectDropdownComponent.prototype, "notFoundMsg", void 0);
    __decorate([
        Input(),
        __metadata("design:type", OptionList)
    ], SelectDropdownComponent.prototype, "optionList", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], SelectDropdownComponent.prototype, "top", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], SelectDropdownComponent.prototype, "width", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SelectDropdownComponent.prototype, "placeholder", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "customClass", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "visibleOptions", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], SelectDropdownComponent.prototype, "dropdownHeight", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], SelectDropdownComponent.prototype, "dropdownMaxHeight", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], SelectDropdownComponent.prototype, "optionHeight", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SelectDropdownComponent.prototype, "enableSelectAll", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "selectAllLabel", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "outline", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "close", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "optionClicked", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "singleFilterClick", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "singleFilterInput", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "singleFilterKeydown", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "animationDone", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "animationStart", void 0);
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "selectAll", void 0);
    __decorate([
        ViewChild('filterInput'),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "filterInput", void 0);
    __decorate([
        ViewChild('optionsList', { static: true }),
        __metadata("design:type", Object)
    ], SelectDropdownComponent.prototype, "optionsList", void 0);
    __decorate([
        ViewChild('dropdownContent', { static: true }),
        __metadata("design:type", ElementRef)
    ], SelectDropdownComponent.prototype, "dropdownContent", void 0);
    __decorate([
        ViewChild('customContent', { static: true }),
        __metadata("design:type", ElementRef)
    ], SelectDropdownComponent.prototype, "customContent", void 0);
    __decorate([
        HostListener('window: keydown', ['$event']),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], SelectDropdownComponent.prototype, "onWindowKeydown", null);
    __decorate([
        HostListener('keyup'),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], SelectDropdownComponent.prototype, "onkeyup", null);
    __decorate([
        HostListener('input'),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], SelectDropdownComponent.prototype, "onkeydown", null);
    SelectDropdownComponent = __decorate([
        Component({
            selector: 'mdb-select-dropdown',
            template: "<div\n  (click)=\"$event.stopPropagation()\"\n  class=\"dropdown-content\"\n  #dropdownContent\n  [ngStyle]=\"{ 'top.px': top, 'left.px': left, 'width.px': width }\"\n  [@dropdownAnimation]=\"{\n    value: state,\n    params: { startHeight: startHeight, endHeight: endHeight }\n  }\"\n  (@dropdownAnimation.done)=\"onAnimationDone()\"\n  (@dropdownAnimation.start)=\"onAnimationStart()\"\n>\n  <div class=\"filter md-form px-2\" *ngIf=\"filterEnabled\">\n    <input\n      type=\"text\"\n      class=\"search form-control w-100 d-block\"\n      #filterInput\n      [attr.autocomplete]=\"filterAutocomplete ? 'on' : 'off'\"\n      [attr.role]=\"'searchbox'\"\n      [placeholder]=\"placeholder\"\n      (input)=\"onSingleFilterInput($event)\"\n      (keydown)=\"onSingleFilterKeydown($event)\"\n    />\n  </div>\n\n  <div class=\"options\" #optionsList>\n    <ul\n      class=\"select-dropdown\"\n      [ngClass]=\"{ 'multiple-select-dropdown': multiple }\"\n      (wheel)=\"onOptionsWheel($event)\"\n    >\n      <li\n        [ngStyle]=\"{ 'height.px': optionHeight }\"\n        *ngIf=\"multiple && enableSelectAll && this.hasOptionsItems\"\n        (click)=\"onSelectAllClick()\"\n      >\n        <span class=\"filtrable\" *ngIf=\"multiple\">\n          <input\n            type=\"checkbox\"\n            [checked]=\"selectAllSelected\"\n            class=\"form-check-input {{ customClass }}\"\n          />\n          <label></label>\n          {{ selectAllLabel }}\n        </span>\n      </li>\n      <li\n        *ngFor=\"let option of optionList.filtered\"\n        [ngClass]=\"{\n          'heavy-rain-gradient': option.highlighted && !highlightColor,\n          active: option.highlighted,\n          selected: option.selected,\n          disabled: option.disabled,\n          optgroup: option.group,\n          'd-flex justify-content-between flex-row-reverse align-items-center': option.icon\n        }\"\n        [ngStyle]=\"{\n          'height.px': optionHeight,\n          'line-height.px': optionHeight,\n          'background-color': getOptionStyle(option)['background-color'],\n          color: getOptionStyle(option)['color']\n        }\"\n        [attr.role]=\"'option'\"\n        [attr.aria-selected]=\"option.selected\"\n        [attr.aria-disabled]=\"option.disabled\"\n        (click)=\"onOptionClick(option)\"\n        (mouseover)=\"option.hovered = true\"\n        (mouseleave)=\"option.hovered = false\"\n      >\n        <img class=\"rounded-circle\" [src]=\"option.icon\" *ngIf=\"option.icon !== ''\" />\n        <span\n          class=\"deselect-option\"\n          *ngIf=\"!multiple\"\n          [ngStyle]=\"{\n            'background-color': getOptionStyle(option)['background-color'],\n            color: getOptionStyle(option)['color']\n          }\"\n          >{{ option.label }}</span\n        >\n        <span\n          class=\"deselect-option\"\n          [ngStyle]=\"{\n            'background-color': getOptionStyle(option)['background-color'],\n            color: getOptionStyle(option)['color']\n          }\"\n          *ngIf=\"multiple\"\n        >\n          <input\n            type=\"checkbox\"\n            [checked]=\"option.selected\"\n            class=\"form-check-input {{ customClass }}\"\n            [disabled]=\"option.disabled\"\n          />\n          <label></label>\n          {{ option.label }}\n        </span>\n      </li>\n      <li\n        *ngIf=\"!this.hasOptionsItems\"\n        class=\"message disabled\"\n        [ngStyle]=\"{ 'height.px': optionHeight }\"\n      >\n        <span>{{ notFoundMsg }}</span>\n      </li>\n      <li #customContent class=\"custom-select-content\">\n        <ng-content></ng-content>\n      </li>\n    </ul>\n  </div>\n</div>\n",
            encapsulation: ViewEncapsulation.None,
            changeDetection: ChangeDetectionStrategy.OnPush,
            animations: [
                trigger('dropdownAnimation', [
                    state('invisible', style({ opacity: 0, height: '0px' })),
                    state('visible', style({ opacity: 1, height: '*' })),
                    transition('invisible => visible', animate('300ms ease')),
                    transition('visible => invisible', animate('300ms ease')),
                ]),
            ]
        }),
        __metadata("design:paramtypes", [ElementRef,
            Renderer2,
            ChangeDetectorRef])
    ], SelectDropdownComponent);
    return SelectDropdownComponent;
}());
export { SelectDropdownComponent };
